diff --git a/bolt/test/AArch64/Inputs/array_end.lld_script b/bolt/test/AArch64/Inputs/array_end.lld_script
deleted file mode 100644
index bf77c0493a09..000000000000
--- a/bolt/test/AArch64/Inputs/array_end.lld_script
+++ /dev/null
@@ -1,15 +0,0 @@
-SECTIONS {
-  .interp : { *(.interp) }
-
-  . = ALIGN(CONSTANT(MAXPAGESIZE));
-  .fini_array    :
-  {
-    PROVIDE_HIDDEN (__fini_array_start = .);
-    KEEP (*(SORT_BY_INIT_PRIORITY(.fini_array.*) SORT_BY_INIT_PRIORITY(.dtors.*)))
-    PROVIDE_HIDDEN (__fini_array_end = .);
-  }
-
-  . = . + 128;
-
-  .text : { *(.text) }
-}
