diff --git a/bolt/tools/bat-dump/CMakeLists.txt b/bolt/tools/bat-dump/CMakeLists.txt
deleted file mode 100644
index 037b7d2662ae..000000000000
--- a/bolt/tools/bat-dump/CMakeLists.txt
+++ /dev/null
@@ -1,15 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Object
-  Support
-  )
-
-add_bolt_executable(llvm-bat-dump
-  bat-dump.cpp
-
-  DISABLE_LLVM_LINK_LLVM_DYLIB
-  )
-
-target_link_libraries(llvm-bat-dump
-  PRIVATE
-  LLVMBOLTProfile
-  )
