diff --git a/bolt/test/AArch64/symbol-hashes.test b/bolt/test/AArch64/symbol-hashes.test
deleted file mode 100644
index 3f2d869846c2..000000000000
--- a/bolt/test/AArch64/symbol-hashes.test
+++ /dev/null
@@ -1,6 +0,0 @@
-// This test checks that we don't try to use symbols outside of section when
-// generating symbol hashes
-
-RUN: yaml2obj %p/Inputs/symbol-hashes.yaml -o %t.exe
-RUN: llvm-bolt %t.exe -force-data-relocations -o %t.exe.bolt
-
