diff --git a/bolt/test/runtime/AArch64/issue177.s b/bolt/test/runtime/AArch64/issue177.s
deleted file mode 100644
index 3ed8d7cbf50c..000000000000
--- a/bolt/test/runtime/AArch64/issue177.s
+++ /dev/null
@@ -1,28 +0,0 @@
-# This reproduces issue 177 from our github repo
-# AARCH64_MOVW_UABS_G* relocations handling
-
-# REQUIRES: system-linux
-
-# RUN: %clang %cflags -no-pie %s -o %t.exe -Wl,-q
-# RUN: llvm-bolt %t.exe -o %t.bolt --use-old-text=0 --lite=0 --trap-old-code
-# RUN: %t.bolt
-
-.text
-.align 4
-.global test
-.type test, %function
-test:
-mov x0, xzr
-ret
-.size test, .-test
-
-.align 4
-.global main
-.type main, %function
-main:
-movz x0, #:abs_g3:test
-movk x0, #:abs_g2_nc:test
-movk x0, #:abs_g1_nc:test
-movk x0, #:abs_g0_nc:test
-br x0
-.size main, .-main
