diff --git a/bolt/test/perf2bolt/lit.local.cfg b/bolt/test/perf2bolt/lit.local.cfg
deleted file mode 100644
index 0fecf913aa98..000000000000
--- a/bolt/test/perf2bolt/lit.local.cfg
+++ /dev/null
@@ -1,5 +0,0 @@
-import shutil
-import subprocess
-
-if shutil.which("perf") is not None and subprocess.run(["perf", "record", "-e", "cycles:u", "-o", "/dev/null", "--", "perf", "--version"], capture_output=True).returncode == 0:
-    config.available_features.add("perf")
