diff --git a/bolt/test/X86/Inputs/relaxed_tc.s b/bolt/test/X86/Inputs/relaxed_tc.s
deleted file mode 100644
index 75055fb82fd1..000000000000
--- a/bolt/test/X86/Inputs/relaxed_tc.s
+++ /dev/null
@@ -1,21 +0,0 @@
-  .globl main
-  .type main, %function
-main:
-	pushq	%rax
-	callq	foo
-	movl	$0x1, %eax
-	popq	%rdx
-	retq
-.size main, .-main
-
-  .globl foo
-  .type foo, %function
-foo:
-	jmp	bar
-.size foo, .-foo
-
-  .globl bar
-  .type bar, %function
-bar:
-	retq
-.size bar, .-bar
