diff --git a/bolt/test/RISCV/reloc-jal.s b/bolt/test/RISCV/reloc-jal.s
deleted file mode 100644
index ce54265fac05..000000000000
--- a/bolt/test/RISCV/reloc-jal.s
+++ /dev/null
@@ -1,20 +0,0 @@
-// RUN: %clang %cflags -o %t %s
-// RUN: llvm-bolt --print-cfg --print-only=_start -o %t.null %t \
-// RUN:    | FileCheck %s
-
-  .text
-
-  .global f
-  .p2align 1
-f:
-  ret
-  .size f, .-f
-
-// CHECK: Binary Function "_start" after building cfg {
-  .globl _start
-  .p2align 1
-_start:
-// CHECK: jal ra, f
-  jal ra, f
-  ret
-  .size _start, .-_start
