diff --git a/bolt/tools/heatmap/CMakeLists.txt b/bolt/tools/heatmap/CMakeLists.txt
deleted file mode 100644
index acddc7a50e8b..000000000000
--- a/bolt/tools/heatmap/CMakeLists.txt
+++ /dev/null
@@ -1,20 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  ${LLVM_TARGETS_TO_BUILD}
-  MC
-  Object
-  Support
-  )
-
-add_bolt_tool(llvm-bolt-heatmap
-  heatmap.cpp
-  DISABLE_LLVM_LINK_LLVM_DYLIB
-  )
-
-target_link_libraries(llvm-bolt-heatmap
-  PRIVATE
-  LLVMBOLTProfile
-  LLVMBOLTRewrite
-  LLVMBOLTUtils
-  )
-
-add_dependencies(bolt llvm-bolt-heatmap)
