diff --git a/bolt/test/RISCV/reloc-branch.s b/bolt/test/RISCV/reloc-branch.s
deleted file mode 100644
index 6a8b5a28e19d..000000000000
--- a/bolt/test/RISCV/reloc-branch.s
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: %clang %cflags -o %t %s
-// RUN: llvm-bolt --print-cfg --print-only=_start -o %t.null %t \
-// RUN:    | FileCheck %s
-
-  .text
-  .globl _start
-  .p2align 1
-// CHECK: Binary Function "_start" after building cfg {
-_start:
-// CHECK: beq zero, zero, .Ltmp0
-  beq zero, zero, 1f
-  nop
-// CHECK: .Ltmp0
-1:
-  ret
-  .size _start, .-_start
