diff --git a/bolt/test/runtime/X86/Inputs/basic-instrumentation.s b/bolt/test/runtime/X86/Inputs/basic-instrumentation.s
deleted file mode 100644
index 388d2cdbd085..000000000000
--- a/bolt/test/runtime/X86/Inputs/basic-instrumentation.s
+++ /dev/null
@@ -1,9 +0,0 @@
-  .globl main
-  .type main, %function
-main:
-	pushq	%rbp
-	movq	%rsp, %rbp
-	movl	$0x0, %eax
-	popq	%rbp
-	retq
-.size main, .-main
