diff --git a/bolt/test/X86/merge-fdata-nobat-mode.test b/bolt/test/X86/merge-fdata-nobat-mode.test
deleted file mode 100644
index 978052e35007..000000000000
--- a/bolt/test/X86/merge-fdata-nobat-mode.test
+++ /dev/null
@@ -1,6 +0,0 @@
-## Check that merge-fdata tool doesn't spuriously print boltedcollection
-
-RUN: merge-fdata %S/Inputs/blarge.fdata %S/Inputs/blarge.fdata \
-RUN:   | FileCheck %s --check-prefix=CHECK-FDATA
-
-CHECK-FDATA-NOT: boltedcollection
