diff --git a/bolt/test/unreadable-profile.test b/bolt/test/unreadable-profile.test
deleted file mode 100644
index fe1ca93f3221..000000000000
--- a/bolt/test/unreadable-profile.test
+++ /dev/null
@@ -1,13 +0,0 @@
-REQUIRES: system-linux
-
-RUN: touch %t.profile && chmod 000 %t.profile
-RUN: %clang %S/Inputs/hello.c -o %t
-RUN: not llvm-bolt %t -o %t.bolt --data %t.profile 2>&1 \
-RUN:   | FileCheck %s --check-prefix CHECK-NOPERM
-RUN: not llvm-bolt %t -o %t.bolt --data %t.fake.profile 2>&1 \
-RUN:   | FileCheck %s --check-prefix CHECK-FAKE
-
-## Check that llvm-bolt gracefully handles errors accessing profile data.
-
-CHECK-NOPERM: Permission denied
-CHECK-FAKE: No such file or directory
