diff --git a/bolt/test/cache+-deprecated.test b/bolt/test/cache+-deprecated.test
deleted file mode 100644
index c320ea8816e7..000000000000
--- a/bolt/test/cache+-deprecated.test
+++ /dev/null
@@ -1,8 +0,0 @@
-
-
-REQUIRES: system-linux
-
-RUN: %clangxx %p/Inputs/bolt_icf.cpp -g -Wl,-q -o %t.exe
-RUN: llvm-bolt %t.exe --reorder-blocks=cache+ --relocs -o %t 2>&1 | FileCheck %s
-
-CHECK: '-reorder-blocks=cache+' is deprecated, please use '-reorder-blocks=ext-tsp' instead
