diff --git a/bolt/test/runtime/X86/internal-call-instrument.test b/bolt/test/runtime/X86/internal-call-instrument.test
deleted file mode 100644
index ebc61cf5a3bf..000000000000
--- a/bolt/test/runtime/X86/internal-call-instrument.test
+++ /dev/null
@@ -1,13 +0,0 @@
-# Verify that instrumentation pass handles internal calls
-
-REQUIRES: x86_64-linux,bolt-runtime
-
-RUN: %clang -O3 %S/Inputs/internalcall-main.c %S/Inputs/internalcall.S -Wl,-q \
-RUN:   -o %t.exe
-RUN: llvm-bolt --instrument %t.exe --instrumentation-file=%t.fdata \
-RUN:   -o %t -relocs -lite=0
-RUN: %t | FileCheck %s
-
-CHECK:   Case 1: Result is: DEADBEEF
-CHECK:   Case 2: Result is: DEADBEEF
-CHECK:   Case 3: Result is: DEADBEEF
