diff --git a/bolt/test/runtime/X86/exceptions-run.test b/bolt/test/runtime/X86/exceptions-run.test
deleted file mode 100644
index 45702ef358a0..000000000000
--- a/bolt/test/runtime/X86/exceptions-run.test
+++ /dev/null
@@ -1,15 +0,0 @@
-# Verify the exceptions handling work after moving code.
-# The input has dummy nops that are removed by bolt.
-
-REQUIRES: x86_64-linux
-
-RUN: %clangxx -g %S/Inputs/exception3.cpp -o %t.exe
-RUN: llvm-bolt --align-blocks=0 %t.exe -o %t && %t 2>&1 | FileCheck %s
-RUN: llvm-bolt --reorder-blocks=reverse --align-blocks=0 %t.exe -o %t && %t 2>&1 \
-RUN:   | FileCheck %s
-
-CHECK: catch 2
-CHECK-NEXT: catch 1
-CHECK-NEXT: caught ExcC
-
-
