diff --git a/bolt/test/AArch64/data-at-0-offset.c b/bolt/test/AArch64/data-at-0-offset.c
deleted file mode 100644
index e0c689a19bed..000000000000
--- a/bolt/test/AArch64/data-at-0-offset.c
+++ /dev/null
@@ -1,17 +0,0 @@
-// RUN: %clang %cflags -O2 -fPIE -Wl,-q -pie  %s -o %t.exe
-// RUN: llvm-bolt %t.exe -o %t.bolt 2>&1 | FileCheck %s
-// CHECK-NOT: BOLT-WARNING: unable to disassemble instruction at offset
-
-void extra_space() {
-  asm volatile(".rept 256\n"
-               "    .byte 0xff\n"
-               ".endr\n");
-  return;
-}
-
-int main(int argc, char **argv) {
-  void (*fn)(void);
-  fn = extra_space + 256;
-  fn();
-  return 0;
-}
