diff --git a/bolt/test/X86/Inputs/sctc_bug.s b/bolt/test/X86/Inputs/sctc_bug.s
deleted file mode 100644
index 19eeb7bd59f6..000000000000
--- a/bolt/test/X86/Inputs/sctc_bug.s
+++ /dev/null
@@ -1,25 +0,0 @@
-.text
-
-.globl	foo
-.type	foo, @function
-foo:
-  ret
-	.size	foo, .-foo
-
-.globl	main
-.type	main, @function
-main:
-	.cfi_startproc
-
-  movsd   8(%rdi), %xmm0
-  ucomisd 8(%rsi), %xmm0
-  jp      .Lend
-  jne     .Lend
-  jmp     foo
-.Lend:
-  xorl    %eax, %eax
-  retq
-
-	.cfi_endproc
-	.size	main, .-main
-
