diff --git a/bolt/unittests/CMakeLists.txt b/bolt/unittests/CMakeLists.txt
deleted file mode 100644
index 64414b83d39f..000000000000
--- a/bolt/unittests/CMakeLists.txt
+++ /dev/null
@@ -1,9 +0,0 @@
-add_custom_target(BoltUnitTests)
-set_target_properties(BoltUnitTests PROPERTIES FOLDER "BOLT/Tests")
-
-function(add_bolt_unittest test_dirname)
-  add_unittest(BoltUnitTests ${test_dirname} ${ARGN})
-endfunction()
-
-add_subdirectory(Core)
-add_subdirectory(Profile)
