diff --git a/bolt/test/runtime/X86/section-order.test b/bolt/test/runtime/X86/section-order.test
deleted file mode 100644
index a1317daba50e..000000000000
--- a/bolt/test/runtime/X86/section-order.test
+++ /dev/null
@@ -1,10 +0,0 @@
-REQUIRES: system-linux,bolt-runtime
-
-RUN: %clang %p/Inputs/basic-instrumentation.s -Wl,-q -o %t.exe
-RUN: llvm-bolt %t.exe -o %t --instrument
-RUN: llvm-readelf --section-headers %t | FileCheck %s
-
-## Verify that llvm-bolt outputs new sections in expected order.
-CHECK: .text.bolt.extra.1
-CHECK: .rodata.bolt.extra.1
-CHECK: .data.bolt.extra.1
