diff --git a/bolt/test/runtime/X86/internal-call.test b/bolt/test/runtime/X86/internal-call.test
deleted file mode 100644
index ecd74e972b8c..000000000000
--- a/bolt/test/runtime/X86/internal-call.test
+++ /dev/null
@@ -1,14 +0,0 @@
-# Verify that we support x86 old pic style internal calls accordingly in
-# relocation mode
-#
-
-REQUIRES: x86_64-linux
-
-RUN: %clang -O3 %S/Inputs/internalcall-main.c %S/Inputs/internalcall.S -Wl,-q \
-RUN:   -o %t.exe
-RUN: llvm-bolt %t.exe -o %t --relocs --lite=0
-RUN: %t | FileCheck %s
-
-CHECK:   Case 1: Result is: DEADBEEF
-CHECK:   Case 2: Result is: DEADBEEF
-CHECK:   Case 3: Result is: DEADBEEF
