diff --git a/bolt/test/AArch64/lit.local.cfg b/bolt/test/AArch64/lit.local.cfg
deleted file mode 100644
index 9432240469c7..000000000000
--- a/bolt/test/AArch64/lit.local.cfg
+++ /dev/null
@@ -1,7 +0,0 @@
-if "AArch64" not in config.root.targets:
-    config.unsupported = True
-
-flags = "--target=aarch64-unknown-linux-gnu -nostartfiles -nostdlib -ffreestanding"
-
-config.substitutions.insert(0, ("%cflags", f"%cflags {flags}"))
-config.substitutions.insert(0, ("%cxxflags", f"%cxxflags {flags}"))
