diff --git a/bolt/test/X86/plt-sec-8-byte.test b/bolt/test/X86/plt-sec-8-byte.test
deleted file mode 100644
index d32faefe7cbb..000000000000
--- a/bolt/test/X86/plt-sec-8-byte.test
+++ /dev/null
@@ -1,6 +0,0 @@
-## Check correctness of processing .plt.sec with 8-byte entries.
-# RUN: yaml2obj %p/Inputs/plt-sec-8-byte.yaml &> %t.exe
-# RUN: llvm-bolt %t.exe --print-cfg --print-only=main -o %t.out | FileCheck %s
-
-CHECK:  callq printf@PLT
-CHECK:  callq exit@PLT
