diff --git a/bolt/test/RISCV/lit.local.cfg b/bolt/test/RISCV/lit.local.cfg
deleted file mode 100644
index ffc9939c98f3..000000000000
--- a/bolt/test/RISCV/lit.local.cfg
+++ /dev/null
@@ -1,7 +0,0 @@
-if 'RISCV' not in config.root.targets:
-    config.unsupported = True
-
-flags = '--target=riscv64 -nostdlib -ffreestanding -Wl,--emit-relocs'
-
-config.substitutions.insert(0, ('%cflags', f'%cflags {flags}'))
-config.substitutions.insert(0, ('%cxxflags', f'%cxxflags {flags}'))
