diff --git a/bolt/test/lit.local.cfg b/bolt/test/lit.local.cfg
deleted file mode 100644
index 8aa5f15d5ccf..000000000000
--- a/bolt/test/lit.local.cfg
+++ /dev/null
@@ -1,6 +0,0 @@
-host_linux_triple = config.target_triple.split("-")[0] + "-unknown-linux-gnu"
-common_linker_flags = "-fuse-ld=lld -Wl,--unresolved-symbols=ignore-all"
-flags = f"--target={host_linux_triple} {common_linker_flags}"
-
-config.substitutions.insert(0, ("%cflags", f"%cflags {flags}"))
-config.substitutions.insert(0, ("%cxxflags", f"%cxxflags {flags}"))
