diff --git a/bolt/test/X86/Inputs/sctc_bug2.s b/bolt/test/X86/Inputs/sctc_bug2.s
deleted file mode 100644
index 782a7478c50b..000000000000
--- a/bolt/test/X86/Inputs/sctc_bug2.s
+++ /dev/null
@@ -1,33 +0,0 @@
-.text
-
-.globl	foo
-.type	foo, @function
-foo:
-  ret
-	.size	foo, .-foo
-
-.globl	main
-.type	main, @function
-main:
-	.cfi_startproc
-
-  cmp   %rdi, 1
-  jne  .L2
-
-  xorl    %eax, %eax
-
-.L1:
-  jmp foo
-
-.L2:
-  jb .L1
-
-  cmp %eax, 0
-  xorl    %eax, %eax
-  ja .L2
-.Lend:
-  retq
-
-	.cfi_endproc
-	.size	main, .-main
-
