diff --git a/bolt/test/X86/Inputs/tailcall_traps.s b/bolt/test/X86/Inputs/tailcall_traps.s
deleted file mode 100644
index 5fb60d94fc4c..000000000000
--- a/bolt/test/X86/Inputs/tailcall_traps.s
+++ /dev/null
@@ -1,18 +0,0 @@
-.globl foo
-foo:
-	.cfi_startproc
-  jmpq	*%rax
-	.cfi_endproc
-.size foo, .-foo
-
-.globl bar
-bar:
-	.cfi_startproc
-  jmp	moo
-	.cfi_endproc
-.size bar, .-bar
-
-.globl moo
-moo:
-  nop
-.size moo, .-moo
