diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/reserved-identifier-c.c b/clang-tools-extra/test/clang-tidy/checkers/bugprone/reserved-identifier-c.c
deleted file mode 100644
index 9d075433ab9e..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/reserved-identifier-c.c
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: %check_clang_tidy %s bugprone-reserved-identifier %t
-
-// in C, double underscores are fine except at the beginning
-
-void foo__(void);
-void f__o__o(void);
-void f_________oo(void);
-void __foo(void);
-// CHECK-MESSAGES: :[[@LINE-1]]:6: warning: declaration uses identifier '__foo', which is a reserved identifier [bugprone-reserved-identifier]
-// CHECK-FIXES: {{^}}void foo(void);{{$}}
