diff --git a/clang-tools-extra/clangd/index/remote/server/CMakeLists.txt b/clang-tools-extra/clangd/index/remote/server/CMakeLists.txt
deleted file mode 100644
index 3bea34826f8a..000000000000
--- a/clang-tools-extra/clangd/index/remote/server/CMakeLists.txt
+++ /dev/null
@@ -1,23 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-add_clang_executable(clangd-index-server
-  Server.cpp
-
-  DEPENDS
-  clangdRemoteIndexProto
-  clangdRemoteIndexServiceProto
-  )
-
-target_link_libraries(clangd-index-server
-  PRIVATE
-  clangDaemon
-  clangdSupport
-
-  clangdMonitoringServiceProto
-  clangdRemoteIndexProto
-  clangdRemoteIndexServiceProto
-  clangdRemoteMarshalling
-
-  ${REFLECTION_LIBRARY}
-  )
