diff --git a/clang-tools-extra/test/modularize/Inputs/SomeOtherTypes.h b/clang-tools-extra/test/modularize/Inputs/SomeOtherTypes.h
deleted file mode 100644
index 288faffbb023..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/SomeOtherTypes.h
+++ /dev/null
@@ -1,4 +0,0 @@
-// Declare another type for the dependency check.
-// This file dependent on SomeTypes.h being included first.
-
-typedef TypeInt OtherTypeInt;
