diff --git a/clang-tools-extra/test/modularize/ProblemsDisplayLists.modularize b/clang-tools-extra/test/modularize/ProblemsDisplayLists.modularize
deleted file mode 100644
index 972156151126..000000000000
--- a/clang-tools-extra/test/modularize/ProblemsDisplayLists.modularize
+++ /dev/null
@@ -1,16 +0,0 @@
-# RUN: not modularize -display-file-lists %S/Inputs/CompileError/module.modulemap 2>&1 | FileCheck %s
-
-# CHECK: {{.*}}{{[/\\]}}Inputs{{[/\\]}}CompileError{{[/\\]}}HasError.h:1:9: error: unknown type name 'WithoutDep'
-
-# CHECK: These are the files with possible errors:
-
-# CHECK: Inputs/CompileError/HasError.h
-
-# CHECK: These are the files with no detected errors:
-
-# CHECK: Inputs/CompileError/Level1A.h
-
-# CHECK: These are the combined files, with problem files preceded by #:
-
-# CHECK:      Inputs/CompileError/HasError.h
-# CHECK-NEXT: Inputs/CompileError/Level1A.h
