diff --git a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/PrivateHeaders/SomeKit_PrivateForNullAnnotation.h b/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/PrivateHeaders/SomeKit_PrivateForNullAnnotation.h
deleted file mode 100644
index bae4456b4080..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/PrivateHeaders/SomeKit_PrivateForNullAnnotation.h
+++ /dev/null
@@ -1,17 +0,0 @@
-#ifndef SOMEKIT_PRIVATE_H
-#define SOMEKIT_PRIVATE_H
-
-#import <SomeKit/SomeKitForNullAnnotation.h>
-
-@interface A(Private)
--(A*)privateTransform:(A*)input;
-
-@property (nonatomic) A* internalProperty;
-@end
-
-@protocol InternalProtocol
-- (id) MomeMethod;
-@end
-
-#endif
-
