diff --git a/clang/include/clang/Basic/arm_bf16.td b/clang/include/clang/Basic/arm_bf16.td
deleted file mode 100644
index f70c7221f8d6..000000000000
--- a/clang/include/clang/Basic/arm_bf16.td
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- arm_bf16.td - ARM BF16 compiler interface ------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-//
-//  This file defines the TableGen definitions from which the ARM BF16 header
-//  file will be generated.
-//
-//===----------------------------------------------------------------------===//
-
-include "arm_neon_incl.td"
