diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-parameters-virtual.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/unused-parameters-virtual.cpp
deleted file mode 100644
index a2bf9d917473..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-parameters-virtual.cpp
+++ /dev/null
@@ -1,20 +0,0 @@
-// RUN: %check_clang_tidy %s misc-unused-parameters %t -- \
-// RUN:   -config="{CheckOptions: {misc-unused-parameters.IgnoreVirtual: true}}" --
-
-struct Base {
-  int f(int foo) {
-  // CHECK-MESSAGES: :[[@LINE-1]]:13: warning: parameter 'foo' is unused [misc-unused-parameters]
-  // CHECK-FIXES: int f(int  /*foo*/) {
-    return 5;
-  }
-
-  virtual int f2(int foo) {
-    return 5;
-  }
-};
-
-struct Derived : Base {
-  int f2(int foo) override {
-    return 5;
-  }
-};
