diff --git a/clang/lib/APINotes/CMakeLists.txt b/clang/lib/APINotes/CMakeLists.txt
deleted file mode 100644
index dc83edd911ce..000000000000
--- a/clang/lib/APINotes/CMakeLists.txt
+++ /dev/null
@@ -1,12 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  BitReader
-  BitstreamReader
-  Support)
-add_clang_library(clangAPINotes
-  APINotesManager.cpp
-  APINotesReader.cpp
-  APINotesTypes.cpp
-  APINotesWriter.cpp
-  APINotesYAMLCompiler.cpp
-  LINK_LIBS
-    clangBasic)
