diff --git a/clang-tools-extra/clangd/tool/ClangdToolMain.cpp b/clang-tools-extra/clangd/tool/ClangdToolMain.cpp
deleted file mode 100644
index b62ff362bd2a..000000000000
--- a/clang-tools-extra/clangd/tool/ClangdToolMain.cpp
+++ /dev/null
@@ -1,13 +0,0 @@
-//===--- ClangdToolMain.cpp - clangd main function ------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#include "ClangdMain.h"
-
-int main(int argc, char **argv) {
-  return clang::clangd::clangdMain(argc, argv);
-}
