diff --git a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/APINotes/SomeKit_private.apinotes b/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/APINotes/SomeKit_private.apinotes
deleted file mode 100644
index 28ede9dfa25c..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/APINotes/SomeKit_private.apinotes
+++ /dev/null
@@ -1,15 +0,0 @@
-Name: SomeKit
-Classes:
-  - Name: A
-    Methods:         
-      - Selector: "privateTransform:input:"
-        MethodKind:      Instance
-        NullabilityOfRet: N
-        Nullability:      [ N, S ]
-    Properties:
-      - Name: internalProperty
-        Nullability: N
-Protocols:
-  - Name: InternalProtocol
-    Availability: none
-    AvailabilityMsg: "not for you"
