diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-file/config-file-spaces b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-file/config-file-spaces
deleted file mode 100644
index 4aa1f846ade6..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-file/config-file-spaces
+++ /dev/null
@@ -1,9 +0,0 @@
-Checks: "
-  -*
-  ,
-  hicpp-uppercase-literal-suffix
-  ,hicpp-use-auto
-
-
-  ,  hicpp-use-emplace
-"
