diff --git a/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/no.cpp b/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/no.cpp
deleted file mode 100644
index 140e266200b7..000000000000
--- a/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/no.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-#include <string>
-// CHECK: #include <string>
-// CHECK-NEXT: #include <memory>
-// CHECK-NEXT: #include "bar.h"
-#include <memory>
-#include "foo.h"
-#include "bar.h"
-
-void foo() {
-}
