diff --git a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstdio b/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstdio
deleted file mode 100644
index 5024529f96da..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstdio
+++ /dev/null
@@ -1,21 +0,0 @@
-//===--- cstdio - Stub header for tests -------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef _STDIO_
-#define _STDIO_
-
-#include <stdio.h>
-
-namespace std {
-  using ::FILE;
-  using ::printf;
-  using ::fprintf;
-}
-
-#endif // _STDIO_
-
