diff --git a/clang-tools-extra/clang-change-namespace/CMakeLists.txt b/clang-tools-extra/clang-change-namespace/CMakeLists.txt
deleted file mode 100644
index ded91edb8e34..000000000000
--- a/clang-tools-extra/clang-change-namespace/CMakeLists.txt
+++ /dev/null
@@ -1,27 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  FrontendOpenMP
-  Support
-  )
-
-add_clang_library(clangChangeNamespace
-  ChangeNamespace.cpp
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-)
-
-clang_target_link_libraries(clangChangeNamespace
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangFormat
-  clangFrontend
-  clangLex
-  clangSerialization
-  clangTooling
-  clangToolingCore
-  )
-
-add_subdirectory(tool)
