diff --git a/clang/test/APINotes/Inputs/Frameworks/LayeredKitImpl.framework/Modules/module.modulemap b/clang/test/APINotes/Inputs/Frameworks/LayeredKitImpl.framework/Modules/module.modulemap
deleted file mode 100644
index 58a6e55c1067..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/LayeredKitImpl.framework/Modules/module.modulemap
+++ /dev/null
@@ -1,5 +0,0 @@
-framework module LayeredKitImpl {
-  umbrella header "LayeredKitImpl.h"
-  export *
-  module * { export * }
-}
