diff --git a/clang-tools-extra/test/modularize/Inputs/CoverageProblems/module.modulemap b/clang-tools-extra/test/modularize/Inputs/CoverageProblems/module.modulemap
deleted file mode 100644
index 2b8b23054a01..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/CoverageProblems/module.modulemap
+++ /dev/null
@@ -1,30 +0,0 @@
-// module.modulemap
-
-module Level1A {
-  header "Level1A.h"
-  export *
-}
-module Level1B {
-  header "Level1B.h"
-  export *
-  module Level2B {
-    header "Level2B.h"
-    export *
-  }
-}
-module Level2A {
-  header "Level2A.h"
-  export *
-}
-module UmbrellaDirectoryModule {
-  umbrella "UmbrellaSub"
-}
-module UmbrellaHeaderModule {
-  umbrella header "UmbrellaFile.h"
-}
-/*
-module NoHeader {
-  header "NoHeader.h"
-  export *
-}
-*/
