diff --git a/clang-tools-extra/test/clang-include-fixer/prefix_variable.cpp b/clang-tools-extra/test/clang-include-fixer/prefix_variable.cpp
deleted file mode 100644
index b39cdb8d38d7..000000000000
--- a/clang-tools-extra/test/clang-include-fixer/prefix_variable.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: sed -e 's#//.*$##' %s > %t.cpp
-// RUN: clang-include-fixer -db=yaml -input=%p/Inputs/fake_yaml_db.yaml %t.cpp --
-// RUN: FileCheck %s -input-file=%t.cpp
-
-// CHECK-NOT: #include
-// CHECK: doesnotexist f;
-
-namespace b {
-doesnotexist f;
-}
