diff --git a/clang/test/APINotes/Inputs/Frameworks/FrameworkWithActualPrivateModule.framework/Modules/module.modulemap b/clang/test/APINotes/Inputs/Frameworks/FrameworkWithActualPrivateModule.framework/Modules/module.modulemap
deleted file mode 100644
index 859d723716be..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/FrameworkWithActualPrivateModule.framework/Modules/module.modulemap
+++ /dev/null
@@ -1,5 +0,0 @@
-framework module FrameworkWithActualPrivateModule {
-  umbrella header "FrameworkWithActualPrivateModule.h"
-  export *
-  module * { export * }
-}
