diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/undelegated-constructor-cxx98.cpp b/clang-tools-extra/test/clang-tidy/checkers/bugprone/undelegated-constructor-cxx98.cpp
deleted file mode 100644
index f859f24dd7ba..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/undelegated-constructor-cxx98.cpp
+++ /dev/null
@@ -1,23 +0,0 @@
-// RUN: clang-tidy %s -checks=-*,bugprone-undelegated-constructor -- -std=c++98 | count 0
-
-// Note: this test expects no diagnostics, but FileCheck cannot handle that,
-// hence the use of | count 0.
-
-struct Ctor;
-Ctor foo();
-
-struct Ctor {
-  Ctor();
-  Ctor(int);
-  Ctor(int, int);
-  Ctor(Ctor *i) {
-    Ctor();
-    Ctor(0);
-    Ctor(1, 2);
-    foo();
-  }
-};
-
-Ctor::Ctor() {
-  Ctor(1);
-}
