diff --git a/clang-tools-extra/test/modularize/Inputs/NestedMacro.h b/clang-tools-extra/test/modularize/Inputs/NestedMacro.h
deleted file mode 100644
index 2cfc92c081b1..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/NestedMacro.h
+++ /dev/null
@@ -1,5 +0,0 @@
-// Verification of fix for nested macro.
-
-#define FUNCMACROINNER(a) a
-#define FUNCMACROOUTER(b, c) FUNCMACROINNER(b) + FUNCMACROINNER(c)
-int FuncMacroValue = FUNCMACROOUTER(1, 2);
