diff --git a/clang-tools-extra/test/modularize/NoProblemsDependencies.modularize b/clang-tools-extra/test/modularize/NoProblemsDependencies.modularize
deleted file mode 100644
index bd5292b8099e..000000000000
--- a/clang-tools-extra/test/modularize/NoProblemsDependencies.modularize
+++ /dev/null
@@ -1,3 +0,0 @@
-# RUN: modularize %s -x c++
-
-Inputs/IsDependent.h: Inputs/SomeTypes.h Inputs/SomeOtherTypes.h
