diff --git a/clang/test/APINotes/Inputs/Headers/Methods.h b/clang/test/APINotes/Inputs/Headers/Methods.h
deleted file mode 100644
index f46fe31533e5..000000000000
--- a/clang/test/APINotes/Inputs/Headers/Methods.h
+++ /dev/null
@@ -1,14 +0,0 @@
-struct IntWrapper {
-  int value;
-
-  IntWrapper getIncremented() const { return {value + 1}; }
-};
-
-// TODO: support nested tags
-struct Outer {
-  struct Inner {
-    int value;
-
-    Inner getDecremented() const { return {value - 1}; }
-  };
-};
