diff --git a/clang/lib/Headers/hlsl.h b/clang/lib/Headers/hlsl.h
deleted file mode 100644
index a9dce4503ddd..000000000000
--- a/clang/lib/Headers/hlsl.h
+++ /dev/null
@@ -1,15 +0,0 @@
-//===----- hlsl.h - HLSL definitions --------------------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef _HLSL_H_
-#define _HLSL_H_
-
-#include "hlsl/hlsl_basic_types.h"
-#include "hlsl/hlsl_intrinsics.h"
-
-#endif //_HLSL_H_
