diff --git a/clang/lib/Analysis/CodeInjector.cpp b/clang/lib/Analysis/CodeInjector.cpp
deleted file mode 100644
index 412de96a13b9..000000000000
--- a/clang/lib/Analysis/CodeInjector.cpp
+++ /dev/null
@@ -1,14 +0,0 @@
-//===-- CodeInjector.cpp ----------------------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#include "clang/Analysis/CodeInjector.h"
-
-using namespace clang;
-
-CodeInjector::CodeInjector() {}
-CodeInjector::~CodeInjector() {}
