diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-consteval-if.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-consteval-if.cpp
deleted file mode 100644
index aa712edf445b..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-consteval-if.cpp
+++ /dev/null
@@ -1,31 +0,0 @@
-// RUN: clang-tidy %s -checks='-*,readability-braces-around-statements' -- -std=c++2b | count 0
-
-constexpr void handle(bool) {}
-
-constexpr void shouldPass() {
-  if consteval {
-    handle(true);
-  } else {
-    handle(false);
-  }
-}
-
-constexpr void shouldPassNegated() {
-  if !consteval {
-    handle(false);
-  } else {
-    handle(true);
-  }
-}
-
-constexpr void shouldPassSimple() {
-  if consteval {
-    handle(true);
-  }
-}
-
-void run() {
-    shouldPass();
-    shouldPassNegated();
-    shouldPassSimple();
-}
