diff --git a/clang-tools-extra/docs/doxygen-mainpage.dox b/clang-tools-extra/docs/doxygen-mainpage.dox
deleted file mode 100644
index a4527b8b2009..000000000000
--- a/clang-tools-extra/docs/doxygen-mainpage.dox
+++ /dev/null
@@ -1,9 +0,0 @@
-/// \mainpage clang-tools
-///
-/// \section main_intro Introduction
-/// Welcome to clang tools.
-///
-/// This documentation describes the **internal** software that makes
-/// up clang tools, not the **external** use of clang tools. For
-/// usage instructions, please see the programmer's guide or reference
-/// manual.
