diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/inc-dec-in-conditions-bitint-no-crash.c b/clang-tools-extra/test/clang-tidy/checkers/bugprone/inc-dec-in-conditions-bitint-no-crash.c
deleted file mode 100644
index cfb64c10fe46..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/inc-dec-in-conditions-bitint-no-crash.c
+++ /dev/null
@@ -1,9 +0,0 @@
-// RUN: %check_clang_tidy %s bugprone-inc-dec-in-conditions %t
-
-_BitInt(8) v_401_0() {
-  0 && ({
-    _BitInt(5) y = 0;
-    16777215wb ?: ++y;
-  });
-}
-// CHECK-MESSAGES: warning 
