diff --git a/clang/lib/InstallAPI/CMakeLists.txt b/clang/lib/InstallAPI/CMakeLists.txt
deleted file mode 100644
index b63173bc1be3..000000000000
--- a/clang/lib/InstallAPI/CMakeLists.txt
+++ /dev/null
@@ -1,23 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  TextAPI
-  TextAPIBinaryReader
-  Demangle
-  Core
-  )
-
-add_clang_library(clangInstallAPI
-  DiagnosticBuilderWrappers.cpp
-  DirectoryScanner.cpp
-  DylibVerifier.cpp
-  FileList.cpp
-  Frontend.cpp
-  HeaderFile.cpp
-  Library.cpp
-  Visitor.cpp
-
-  LINK_LIBS
-  clangAST
-  clangBasic
-  clangLex
-  )
