diff --git a/clang/test/APINotes/Inputs/Headers/Templates.h b/clang/test/APINotes/Inputs/Headers/Templates.h
deleted file mode 100644
index 862035fee363..000000000000
--- a/clang/test/APINotes/Inputs/Headers/Templates.h
+++ /dev/null
@@ -1,9 +0,0 @@
-template <typename T>
-struct Box {
-  T value;
-
-  const T& get_value() const { return value; }
-  const T* get_ptr() const { return &value; }
-};
-
-using IntBox = Box<int>;
