diff --git a/clang-tools-extra/test/pp-trace/Inputs/module.modulemap b/clang-tools-extra/test/pp-trace/Inputs/module.modulemap
deleted file mode 100644
index 415c874f09d3..000000000000
--- a/clang-tools-extra/test/pp-trace/Inputs/module.modulemap
+++ /dev/null
@@ -1,18 +0,0 @@
-// module.modulemap
-
-module Level1A {
-  header "Level1A.h"
-  export *
-}
-module Level1B {
-  header "Level1B.h"
-  export *
-  module Level2B {
-    header "Level2B.h"
-    export *
-  }
-}
-module Level2A {
-  header "Level2A.h"
-  export *
-}
