diff --git a/clang/lib/Analysis/plugins/SampleAnalyzer/CMakeLists.txt b/clang/lib/Analysis/plugins/SampleAnalyzer/CMakeLists.txt
deleted file mode 100644
index 77acc47fd483..000000000000
--- a/clang/lib/Analysis/plugins/SampleAnalyzer/CMakeLists.txt
+++ /dev/null
@@ -1,13 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-
-set(LLVM_EXPORTED_SYMBOL_FILE ${CMAKE_CURRENT_SOURCE_DIR}/SampleAnalyzerPlugin.exports)
-add_llvm_library(SampleAnalyzerPlugin MODULE BUILDTREE_ONLY MainCallChecker.cpp)
-
-clang_target_link_libraries(SampleAnalyzerPlugin PRIVATE
-  clangAnalysis
-  clangAST
-  clangStaticAnalyzerCore
-  clangStaticAnalyzerFrontend
-  )
