diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/duplicate-reports.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/duplicate-reports.cpp
deleted file mode 100644
index c9d48d68a4c6..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/duplicate-reports.cpp
+++ /dev/null
@@ -1,14 +0,0 @@
-// RUN: %check_clang_tidy %s cert-err09-cpp,cert-err61-cpp %t -- -- -fexceptions
-
-void alwaysThrows() {
-  int ex = 42;
-  // CHECK-MESSAGES: warning: throw expression should throw anonymous temporary values instead [cert-err09-cpp,cert-err61-cpp]
-  throw ex;
-}
-
-void doTheJob() {
-  try {
-    alwaysThrows();
-  } catch (int&) {
-  }
-}
