diff --git a/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/transitive2.h b/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/transitive2.h
deleted file mode 100644
index ea567885f348..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/transitive2.h
+++ /dev/null
@@ -1,2 +0,0 @@
-#include <s.h>
-#include <t.h>
