diff --git a/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSubDefined.h b/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSubDefined.h
deleted file mode 100644
index 5f0cc6236f62..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSubDefined.h
+++ /dev/null
@@ -1,9 +0,0 @@
-#if !defined(_HEADERGUARDSUBSUBDEFINED_H_)
-#define _HEADERGUARDSUBSUBDEFINED_H_
-
-#define SOMETHING_OTHER 1
-
-// Nest include.  Header guard should not confuse modularize.
-#include "HeaderGuard.h"
-
-#endif // _HEADERGUARDSUBSUBDEFINED_H_
