diff --git a/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSub.h b/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSub.h
deleted file mode 100644
index 3f230d2ebe38..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/HeaderGuardSubSub.h
+++ /dev/null
@@ -1,9 +0,0 @@
-#ifndef _HEADERGUARDSUBSUB_H_
-#define _HEADERGUARDSUBSUB_H_
-
-#define SOMETHING 1
-
-// Nest include.  Header guard should not confuse modularize.
-#include "HeaderGuard.h"
-
-#endif // _HEADERGUARDSUBSUB_H_
