diff --git a/clang/include/clang/Frontend/FrontendDiagnostic.h b/clang/include/clang/Frontend/FrontendDiagnostic.h
deleted file mode 100644
index f41504d8026d..000000000000
--- a/clang/include/clang/Frontend/FrontendDiagnostic.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- DiagnosticFrontend.h - Diagnostics for frontend --------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_FRONTEND_FRONTENDDIAGNOSTIC_H
-#define LLVM_CLANG_FRONTEND_FRONTENDDIAGNOSTIC_H
-
-#include "clang/Basic/DiagnosticFrontend.h"
-
-#endif
