diff --git a/clang-tools-extra/clangd/index/remote/monitor/CMakeLists.txt b/clang-tools-extra/clangd/index/remote/monitor/CMakeLists.txt
deleted file mode 100644
index 194c7d5128a9..000000000000
--- a/clang-tools-extra/clangd/index/remote/monitor/CMakeLists.txt
+++ /dev/null
@@ -1,18 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-add_clang_executable(clangd-index-server-monitor
-  Monitor.cpp
-
-  DEPENDS
-  clangdRemoteIndexServiceProto
-  )
-
-target_link_libraries(clangd-index-server-monitor
-  PRIVATE
-  clangBasic
-  clangdSupport
-
-  clangdMonitoringServiceProto
-  clangdRemoteIndexServiceProto
-  )
