diff --git a/clang/lib/ASTMatchers/Dynamic/CMakeLists.txt b/clang/lib/ASTMatchers/Dynamic/CMakeLists.txt
deleted file mode 100644
index ea1bee407b00..000000000000
--- a/clang/lib/ASTMatchers/Dynamic/CMakeLists.txt
+++ /dev/null
@@ -1,21 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  FrontendOpenMP
-  Support
-)
-
-add_clang_library(clangDynamicASTMatchers
-  Diagnostics.cpp
-  Marshallers.cpp
-  Parser.cpp
-  Registry.cpp
-  VariantValue.cpp
-
-  LINK_LIBS
-  clangAST
-  clangASTMatchers
-  clangBasic
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
