diff --git a/clang-tools-extra/test/modularize/NoProblems.modularize b/clang-tools-extra/test/modularize/NoProblems.modularize
deleted file mode 100644
index b3cc36b605aa..000000000000
--- a/clang-tools-extra/test/modularize/NoProblems.modularize
+++ /dev/null
@@ -1,6 +0,0 @@
-# RUN: modularize %s -x c++
-# RUN: modularize -prefix=%p %s -x c++
-# RUN: modularize -no-coverage-check %S/Inputs/NoProblems.modulemap -x c++
-
-Inputs/SomeTypes.h
-Inputs/SomeDecls.h
