diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-nullptr.c b/clang-tools-extra/test/clang-tidy/checkers/modernize/use-nullptr.c
deleted file mode 100644
index 1218b837199c..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-nullptr.c
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: clang-tidy %s -checks=-*,modernize-use-nullptr -- -std=c17 | count 0
-
-// Note: this test expects no diagnostics, but FileCheck cannot handle that,
-// hence the use of | count 0.
-
-#define NULL 0
-void f(void) {
-  char *str = NULL; // ok
-  (void)str;
-}
