diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/isolate-declaration-no-infinite-loop.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/isolate-declaration-no-infinite-loop.cpp
deleted file mode 100644
index 78a8ec738800..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/isolate-declaration-no-infinite-loop.cpp
+++ /dev/null
@@ -1,7 +0,0 @@
-// RUN: %check_clang_tidy -expect-clang-tidy-error %s readability-isolate-declaration %t
-
-int main(){
-  int a, b
-  // CHECK-MESSAGES: [[@LINE-1]]:3: warning: multiple declarations in a single statement reduces readability
-  // CHECK-MESSAGES: [[@LINE-2]]:11: error: expected ';' at end of declaration [clang-diagnostic-error]
-}
