diff --git a/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/system/transitive.h b/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/system/transitive.h
deleted file mode 100644
index 6dba769fb479..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/portability/Inputs/restrict-system-includes/system/transitive.h
+++ /dev/null
@@ -1,3 +0,0 @@
-#include <r.h>
-#include <t.h>
-#include <s.h>
