diff --git a/clang-tools-extra/test/clang-move/Inputs/type_alias.h b/clang-tools-extra/test/clang-move/Inputs/type_alias.h
deleted file mode 100644
index 7a900f698652..000000000000
--- a/clang-tools-extra/test/clang-move/Inputs/type_alias.h
+++ /dev/null
@@ -1,11 +0,0 @@
-typedef int Int1;
-using Int2 = int;
-
-template<class T>
-struct A {};
-
-template <class T> using B = A<T>;
-
-class C {
-  typedef int Int3;
-};
