diff --git a/clang-tools-extra/test/modularize/Inputs/Anonymous.h b/clang-tools-extra/test/modularize/Inputs/Anonymous.h
deleted file mode 100644
index 8388ecaafd4b..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/Anonymous.h
+++ /dev/null
@@ -1,11 +0,0 @@
-// Exercise some anonymous type issues.
-
-// Anonymous enum.
-enum {
-  Tag1
-};
-
-// Anonymous enum typedef.
-typedef enum {
-  Tag2
-} AnonymousEnum;
