diff --git a/clang-tools-extra/test/clang-query/invalid-command-line.cpp b/clang-tools-extra/test/clang-query/invalid-command-line.cpp
deleted file mode 100644
index e3e8af1d5e7a..000000000000
--- a/clang-tools-extra/test/clang-query/invalid-command-line.cpp
+++ /dev/null
@@ -1,4 +0,0 @@
-// RUN: not clang-query --invalid-arg 2>&1 | FileCheck %s
-
-// CHECK: error: clang-query{{(\.exe)?}}: Unknown command line argument '--invalid-arg'.  Try: 'clang-query{{(\.exe)?}} --help'
-// CHECK-NEXT: clang-query{{(\.exe)?}}: Did you mean '--extra-arg'?
