diff --git a/clang-tools-extra/clang-include-fixer/find-all-symbols/tool/CMakeLists.txt b/clang-tools-extra/clang-include-fixer/find-all-symbols/tool/CMakeLists.txt
deleted file mode 100644
index e6926a0d5bd1..000000000000
--- a/clang-tools-extra/clang-include-fixer/find-all-symbols/tool/CMakeLists.txt
+++ /dev/null
@@ -1,24 +0,0 @@
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/..)
-
-add_clang_tool(find-all-symbols
-  FindAllSymbolsMain.cpp
-  )
-
-clang_target_link_libraries(find-all-symbols
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangFrontend
-  clangLex
-  clangSerialization
-  clangTooling
-  )
-target_link_libraries(find-all-symbols
-  PRIVATE
-  findAllSymbols
-  )
-
-install(PROGRAMS run-find-all-symbols.py
-  DESTINATION "${CMAKE_INSTALL_DATADIR}/clang"
-  COMPONENT find-all-symbols)
