diff --git a/clang-tools-extra/clangd/test/index-serialization/Inputs/sample.cpp b/clang-tools-extra/clangd/test/index-serialization/Inputs/sample.cpp
deleted file mode 100644
index f9f13b930d62..000000000000
--- a/clang-tools-extra/clangd/test/index-serialization/Inputs/sample.cpp
+++ /dev/null
@@ -1,8 +0,0 @@
-// Include a file to ensure we have multiple sources.
-#include "sample.h"
-
-// This introduces a symbol, a reference and a relation.
-struct Bar : public Foo {
-  // This introduces an OverriddenBy relation by implementing Foo::Func.
-  void Func() override {}
-};
