diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/use-internal-linkage/func.h b/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/use-internal-linkage/func.h
deleted file mode 100644
index 0f2b576a126c..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/use-internal-linkage/func.h
+++ /dev/null
@@ -1,5 +0,0 @@
-#pragma once
-
-void func_header();
-
-#include "func_h.inc"
