diff --git a/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/yes.cpp b/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/yes.cpp
deleted file mode 100644
index 20e6b90c39b6..000000000000
--- a/clang-tools-extra/test/clang-apply-replacements/Inputs/format_header/yes.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-#include <string>
-// CHECK: #include "bar.h"
-// CHECK-NEXT: #include <memory>
-// CHECK-NEXT: #include <string>
-#include <memory>
-#include "foo.h"
-#include "bar.h"
-
-void foo() {
-}
