diff --git a/clang-tools-extra/test/clang-query/Inputs/runtime_file.script b/clang-tools-extra/test/clang-query/Inputs/runtime_file.script
deleted file mode 100644
index 714d7f03b1bf..000000000000
--- a/clang-tools-extra/test/clang-query/Inputs/runtime_file.script
+++ /dev/null
@@ -1,5 +0,0 @@
-set bind-root false
-
-l func functionDecl(hasName("bar"))
-m func.bind("f")
-m varDecl().bind("v")
\ No newline at end of file
