diff --git a/clang-tools-extra/test/clang-tidy/checkers/performance/unnecessary-value-param-arc.mm b/clang-tools-extra/test/clang-tidy/checkers/performance/unnecessary-value-param-arc.mm
deleted file mode 100644
index 8e07cedb7d7c..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/performance/unnecessary-value-param-arc.mm
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: clang-tidy %s -checks=-*,performance-unnecessary-value-param -- \
-// RUN:   -xobjective-c++ -fobjc-abi-version=2 -fobjc-arc | count 0
-
-#if !__has_feature(objc_arc)
-#error Objective-C ARC not enabled as expected
-#endif
-
-// Passing an Objective-C ARC-managed object to a C function should
-// not raise performance-unnecessary-value-param.
-void foo(id object) { }
-
-// Same for explicitly non-ARC-managed Objective-C objects.
-void bar(__unsafe_unretained id object) { }
-
-// Same for Objective-c classes.
-void baz(Class c) { }
