diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/Inputs/argument-comment/system-header-with-decl.h b/clang-tools-extra/test/clang-tidy/checkers/bugprone/Inputs/argument-comment/system-header-with-decl.h
deleted file mode 100644
index 4e3529a6f1ab..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/Inputs/argument-comment/system-header-with-decl.h
+++ /dev/null
@@ -1,3 +0,0 @@
-#pragma clang system_header
-
-void my_system_header_function(int arg);
