diff --git a/clang-tools-extra/clangd/Features.inc.in b/clang-tools-extra/clangd/Features.inc.in
deleted file mode 100644
index a43a6ec3a172..000000000000
--- a/clang-tools-extra/clangd/Features.inc.in
+++ /dev/null
@@ -1,7 +0,0 @@
-// IWYU pragma: private, include "Feature.h"
-#define CLANGD_BUILD_XPC @CLANGD_BUILD_XPC@
-#define CLANGD_ENABLE_REMOTE @CLANGD_ENABLE_REMOTE@
-#define ENABLE_GRPC_REFLECTION @ENABLE_GRPC_REFLECTION@
-#define CLANGD_MALLOC_TRIM @CLANGD_MALLOC_TRIM@
-#define CLANGD_TIDY_CHECKS @CLANGD_TIDY_CHECKS@
-#define CLANGD_DECISION_FOREST @CLANGD_DECISION_FOREST@
