diff --git a/clang-tools-extra/test/clang-tidy/checkers/portability/restrict-system-includes-glob.cpp b/clang-tools-extra/test/clang-tidy/checkers/portability/restrict-system-includes-glob.cpp
deleted file mode 100644
index fed902630e63..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/portability/restrict-system-includes-glob.cpp
+++ /dev/null
@@ -1,11 +0,0 @@
-// RUN: %check_clang_tidy %s portability-restrict-system-includes %t \
-// RUN:     -- -config="{CheckOptions: {portability-restrict-system-includes.Includes: '-*,std*.h'}}" \
-// RUN:     -- -isystem %S/Inputs/restrict-system-includes/system
-
-// Test glob functionality: disallow all headers except those that match
-// pattern "std*.h".
-
-#include <stddef.h>
-#include <stdint.h>
-#include <float.h>
-// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: system include float.h not allowed
