diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/include-cleaner.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/include-cleaner.cpp
deleted file mode 100644
index e10ac3f46e2e..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/include-cleaner.cpp
+++ /dev/null
@@ -1,17 +0,0 @@
-// RUN: %check_clang_tidy %s misc-include-cleaner %t -- -- -I%S/Inputs -isystem%S/Inputs/system
-#include "bar.h"
-// CHECK-FIXES: {{^}}#include "baz.h"{{$}}
-#include "foo.h"
-// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: included header foo.h is not used directly [misc-include-cleaner]
-// CHECK-FIXES: {{^}}
-// CHECK-FIXES: {{^}}#include <string>{{$}}
-#include <vector.h>
-// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: included header vector.h is not used directly [misc-include-cleaner]
-// CHECK-FIXES: {{^}}
-int BarResult = bar();
-int BazResult = baz();
-// CHECK-MESSAGES: :[[@LINE-1]]:17: warning: no header providing "baz" is directly included [misc-include-cleaner]
-std::string HelloString;
-// CHECK-MESSAGES: :[[@LINE-1]]:6: warning: no header providing "std::string" is directly included [misc-include-cleaner]
-int FooBarResult = foobar();
-// CHECK-MESSAGES: :[[@LINE-1]]:20: warning: no header providing "foobar" is directly included [misc-include-cleaner]
