diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-assert-failure.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-assert-failure.cpp
deleted file mode 100644
index 7d619c0ba592..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-assert-failure.cpp
+++ /dev/null
@@ -1,12 +0,0 @@
-// RUN: not clang-tidy -checks='-*,readability-braces-around-statements' %s --
-
-// Note: this test expects no assert failure happened in clang-tidy.
-
-int test_failure() {
-  if (std::rand()) {
-  }
-}
-
-void test_failure2() {
-  for (a b c;;
-}
