diff --git a/clang-tools-extra/clangd/xpc/test-client/CMakeLists.txt b/clang-tools-extra/clangd/xpc/test-client/CMakeLists.txt
deleted file mode 100644
index 94477e3bc57e..000000000000
--- a/clang-tools-extra/clangd/xpc/test-client/CMakeLists.txt
+++ /dev/null
@@ -1,25 +0,0 @@
-add_clang_tool(
-  clangd-xpc-test-client
-  ClangdXPCTestClient.cpp
-
-  DEPENDS ClangdXPC
-)
-
-set(LLVM_LINK_COMPONENTS
-    support
-)
-
-clang_target_link_libraries(clangd-xpc-test-client
-  PRIVATE
-  clangBasic
-  clangFormat
-  clangFrontend
-  clangSema
-  clangTooling
-  clangToolingCore
-)
-target_link_libraries(clangd-xpc-test-client
-  PRIVATE
-  clangDaemon
-  clangdXpcJsonConversions
-)
