diff --git a/clang-tools-extra/include-cleaner/test/html.cpp b/clang-tools-extra/include-cleaner/test/html.cpp
deleted file mode 100644
index 127ad54c923f..000000000000
--- a/clang-tools-extra/include-cleaner/test/html.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: clang-include-cleaner -html=- %s -- -I %S/Inputs | FileCheck %s
-#include "bar.h"
-#include "foo.h"
-
-int n = foo();
-// CHECK: <span class='ref sel' data-hover='t{{[0-9]+}}'>foo</span>()
