diff --git a/clang/lib/Frontend/Rewrite/CMakeLists.txt b/clang/lib/Frontend/Rewrite/CMakeLists.txt
deleted file mode 100644
index 64aa651394df..000000000000
--- a/clang/lib/Frontend/Rewrite/CMakeLists.txt
+++ /dev/null
@@ -1,26 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-
-add_clang_library(clangRewriteFrontend
-  FixItRewriter.cpp
-  FrontendActions.cpp
-  HTMLPrint.cpp
-  InclusionRewriter.cpp
-  RewriteMacros.cpp
-  RewriteModernObjC.cpp
-  RewriteObjC.cpp
-  RewriteTest.cpp
-
-  LINK_LIBS
-  clangAST
-  clangBasic
-  clangEdit
-  clangFrontend
-  clangLex
-  clangRewrite
-  clangSerialization
-
-  DEPENDS
-  omp_gen
-  )
