diff --git a/clang/include/clang/CIR/Dialect/IR/CIRDialect.h b/clang/include/clang/CIR/Dialect/IR/CIRDialect.h
deleted file mode 100644
index d53e5d1663d6..000000000000
--- a/clang/include/clang/CIR/Dialect/IR/CIRDialect.h
+++ /dev/null
@@ -1,16 +0,0 @@
-//===- CIRDialect.h - CIR dialect -------------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-//
-// This file declares the CIR dialect.
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_CIR_DIALECT_IR_CIRDIALECT_H
-#define LLVM_CLANG_CIR_DIALECT_IR_CIRDIALECT_H
-
-#endif // LLVM_CLANG_CIR_DIALECT_IR_CIRDIALECT_H
