diff --git a/clang-tools-extra/test/clang-include-fixer/query_symbol.cpp b/clang-tools-extra/test/clang-include-fixer/query_symbol.cpp
deleted file mode 100644
index 84c3a2373bd9..000000000000
--- a/clang-tools-extra/test/clang-include-fixer/query_symbol.cpp
+++ /dev/null
@@ -1,13 +0,0 @@
-// RUN: clang-include-fixer -db=fixed -input='foo= "foo.h","bar.h"' -query-symbol="foo" test.cpp -- | FileCheck %s
-
-// CHECK:     "FilePath": "test.cpp",
-// CHECK-NEXT:"QuerySymbolInfos": [
-// CHECK-NEXT:   {"RawIdentifier": "foo",
-// CHECK-NEXT:    "Range":{"Offset":0,"Length":0}}
-// CHECK-NEXT:],
-// CHECK-NEXT:"HeaderInfos": [
-// CHECK-NEXT:  {"Header": "\"foo.h\"",
-// CHECK-NEXT:   "QualifiedName": "foo"},
-// CHECK-NEXT:  {"Header": "\"bar.h\"",
-// CHECK-NEXT:   "QualifiedName": "foo"}
-// CHECK-NEXT:]
