diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/deduplication.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/deduplication.cpp
deleted file mode 100644
index 056fe4e5735d..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/deduplication.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: %check_clang_tidy %s google-explicit-constructor %t
-
-template<typename T>
-struct A { A(T); };
-// CHECK-MESSAGES: :[[@LINE-1]]:12: warning: single-argument constructors must be marked explicit
-
-void f() {
-  A<int> a(0);
-  A<double> b(0);
-}
