diff --git a/clang-tools-extra/test/clang-tidy/checkers/altera/struct-pack-align-no-crash.cpp b/clang-tools-extra/test/clang-tidy/checkers/altera/struct-pack-align-no-crash.cpp
deleted file mode 100644
index 660addcbe803..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/altera/struct-pack-align-no-crash.cpp
+++ /dev/null
@@ -1,7 +0,0 @@
-// RUN: %check_clang_tidy -expect-clang-tidy-error %s altera-struct-pack-align %t -- -header-filter=.*
-
-struct A;
-struct B {
-  A a;
-// CHECK-MESSAGES: :[[@LINE-1]]:5: error: field has incomplete type 'A' [clang-diagnostic-error]
-};
