diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-objc.m b/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-objc.m
deleted file mode 100644
index 99dd48c4c5ac..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-objc.m
+++ /dev/null
@@ -1,12 +0,0 @@
-// RUN: %check_clang_tidy %s readability-identifier-naming %t \
-// RUN: -config='{CheckOptions: \
-// RUN:  {readability-identifier-naming.ObjcIvarPrefix: '_'}}' \
-// RUN: --
-
-@interface Foo {
-    int _bar;
-    int barWithoutPrefix;
-    // CHECK-MESSAGES: :[[@LINE-1]]:9: warning: invalid case style for objc ivar 'barWithoutPrefix' [readability-identifier-naming]
-    // CHECK-FIXES: int _barWithoutPrefix;
-}
-@end 
