diff --git a/clang-tools-extra/test/clang-move/Inputs/template_class_test.cpp b/clang-tools-extra/test/clang-move/Inputs/template_class_test.cpp
deleted file mode 100644
index 8ad2dcb1eda8..000000000000
--- a/clang-tools-extra/test/clang-move/Inputs/template_class_test.cpp
+++ /dev/null
@@ -1,13 +0,0 @@
-#include "template_class_test.h"
-
-template <typename T>
-void A<T>::g() {}
-
-template <typename T>
-template <typename U>
-void A<T>::k() {}
-
-template <typename T>
-int A<T>::c = 2;
-
-void B::f() {}
