diff --git a/clang/include/clang/Frontend/PCHContainerOperations.h b/clang/include/clang/Frontend/PCHContainerOperations.h
deleted file mode 100644
index 098d32ec3869..000000000000
--- a/clang/include/clang/Frontend/PCHContainerOperations.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- Frontend/PCHContainerOperations.h - PCH Containers -----*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_FRONTEND_PCHCONTAINEROPERATIONS_H
-#define LLVM_CLANG_FRONTEND_PCHCONTAINEROPERATIONS_H
-
-#include "clang/Serialization/PCHContainerOperations.h"
-
-#endif
