diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-precxx11.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-precxx11.cpp
deleted file mode 100644
index 7dc939955f37..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-precxx11.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: %check_clang_tidy -std=c++98,c++03 %s misc-unconventional-assign-operator %t
-
-struct BadArgument {
-  BadArgument &operator=(BadArgument &);
-  // CHECK-MESSAGES: :[[@LINE-1]]:3: warning: operator=() should take 'BadArgument const&' or 'BadArgument'
-};
