diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-bugfix.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-bugfix.cpp
deleted file mode 100644
index 25f2c71a316d..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming-bugfix.cpp
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: %check_clang_tidy -expect-clang-tidy-error %s readability-identifier-naming %t
-
-// This used to cause a null pointer dereference.
-auto [left] = right;
-// CHECK-MESSAGES: :[[@LINE-1]]:15: error: use of undeclared identifier 'right'
-
-namespace crash_on_nonidentifiers {
-struct Foo {
-  operator bool();
-};
-void foo() {
-  // Make sure we don't crash on non-identifier names (e.g. conversion
-  // operators).
-  if (Foo()) {}
-}
-}
