diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.first.hpp b/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.first.hpp
deleted file mode 100644
index a118927647db..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.first.hpp
+++ /dev/null
@@ -1,2 +0,0 @@
-#pragma once
-#include "header-include-cycle.second.hpp"
