diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/assignment-in-if-condition-cxx20.cpp b/clang-tools-extra/test/clang-tidy/checkers/bugprone/assignment-in-if-condition-cxx20.cpp
deleted file mode 100644
index b332b2e49ec7..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/assignment-in-if-condition-cxx20.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: %check_clang_tidy -std=c++20 %s bugprone-assignment-in-if-condition %t
-
-void testRequires() {
-  if constexpr (requires(int &a) { a = 0; }) {
-  }
-}
