diff --git a/clang/lib/StaticAnalyzer/Frontend/CMakeLists.txt b/clang/lib/StaticAnalyzer/Frontend/CMakeLists.txt
deleted file mode 100644
index 0326798e3a17..000000000000
--- a/clang/lib/StaticAnalyzer/Frontend/CMakeLists.txt
+++ /dev/null
@@ -1,30 +0,0 @@
-include_directories( ${CMAKE_CURRENT_BINARY_DIR}/../Checkers )
-
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-
-add_clang_library(clangStaticAnalyzerFrontend
-  AnalysisConsumer.cpp
-  AnalyzerHelpFlags.cpp
-  CheckerRegistry.cpp
-  CreateCheckerManager.cpp
-  FrontendActions.cpp
-  ModelConsumer.cpp
-  ModelInjector.cpp
-
-  LINK_LIBS
-  clangAST
-  clangASTMatchers
-  clangAnalysis
-  clangBasic
-  clangCrossTU
-  clangFrontend
-  clangLex
-  clangStaticAnalyzerCheckers
-  clangStaticAnalyzerCore
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
