diff --git a/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/pro-type-member-init.ignorearrays.cpp b/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/pro-type-member-init.ignorearrays.cpp
deleted file mode 100644
index 01859b3ad98f..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/pro-type-member-init.ignorearrays.cpp
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: %check_clang_tidy %s \
-// RUN: cppcoreguidelines-pro-type-member-init %t \
-// RUN: -config="{CheckOptions: \
-// RUN: {cppcoreguidelines-pro-type-member-init.IgnoreArrays: true}}"
-
-typedef int TypedefArray[4];
-using UsingArray = int[4];
-
-struct HasArrayMember {
-  HasArrayMember() {}
-  // CHECK-MESSAGES: warning: constructor does not initialize these fields: Number
-  UsingArray U;
-  TypedefArray T;
-  int RawArray[4];
-  int Number;
-};
