diff --git a/clang-tools-extra/clang-change-namespace/tool/CMakeLists.txt b/clang-tools-extra/clang-change-namespace/tool/CMakeLists.txt
deleted file mode 100644
index c168bb4d5794..000000000000
--- a/clang-tools-extra/clang-change-namespace/tool/CMakeLists.txt
+++ /dev/null
@@ -1,26 +0,0 @@
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/..)
-
-set(LLVM_LINK_COMPONENTS
-  FrontendOpenMP
-  Support
-  )
-
-add_clang_tool(clang-change-namespace
-  ClangChangeNamespace.cpp
-  )
-clang_target_link_libraries(clang-change-namespace
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangFormat
-  clangFrontend
-  clangRewrite
-  clangSerialization
-  clangTooling
-  clangToolingCore
-  )
-target_link_libraries(clang-change-namespace
-  PRIVATE
-  clangChangeNamespace
-  )
