diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/pr37091.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/pr37091.cpp
deleted file mode 100644
index 6772e9d29f34..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/pr37091.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-// REQUIRES: shell
-// RUN: rm -rf %t
-// RUN: mkdir -p %t
-
-// This is a reproducer for PR37091.
-//
-// Verify that no temporary files are left behind by the clang-tidy invocation.
-
-// RUN: env TMPDIR=%t TEMP=%t TMP=%t clang-tidy -checks='-*,clang-analyzer-core.NullDereference' %s -- --target=mips64
-// RUN: rmdir %t
