diff --git a/clang/test/APINotes/Inputs/Headers/SwiftImportAs.apinotes b/clang/test/APINotes/Inputs/Headers/SwiftImportAs.apinotes
deleted file mode 100644
index b0eead42869a..000000000000
--- a/clang/test/APINotes/Inputs/Headers/SwiftImportAs.apinotes
+++ /dev/null
@@ -1,13 +0,0 @@
----
-Name: SwiftImportAs
-Tags:
-- Name: ImmortalRefType
-  SwiftImportAs: reference
-- Name: RefCountedType
-  SwiftImportAs: reference
-  SwiftReleaseOp: RCRelease
-  SwiftRetainOp: RCRetain
-- Name: NonCopyableType
-  SwiftCopyable: false
-- Name: CopyableType
-  SwiftCopyable: true
