diff --git a/clang/lib/CIR/Dialect/IR/CIRDialect.cpp b/clang/lib/CIR/Dialect/IR/CIRDialect.cpp
deleted file mode 100644
index c2829c3ff2af..000000000000
--- a/clang/lib/CIR/Dialect/IR/CIRDialect.cpp
+++ /dev/null
@@ -1,13 +0,0 @@
-//===- CIRDialect.cpp - MLIR CIR ops implementation -----------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-//
-// This file implements the CIR dialect and its operations.
-//
-//===----------------------------------------------------------------------===//
-
-#include <clang/CIR/Dialect/IR/CIRDialect.h>
