diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/file-filter/system/system-header.h b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/file-filter/system/system-header.h
deleted file mode 100644
index 6e338c85f394..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/file-filter/system/system-header.h
+++ /dev/null
@@ -1,3 +0,0 @@
-class A0 { A0(int); };
-
-#define TO_FLOAT_PTR(x) ((float *)(x))
