diff --git a/clang/test/APINotes/yaml-parse-diags.c b/clang/test/APINotes/yaml-parse-diags.c
deleted file mode 100644
index 3ae39ccb301d..000000000000
--- a/clang/test/APINotes/yaml-parse-diags.c
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: rm -rf %t
-// RUN: %clang_cc1 -fsyntax-only -fapinotes %s -I %S/Inputs/BrokenHeaders -verify
-
-#include "SomeBrokenLib.h"
-
-// expected-error@APINotes.apinotes:4{{unknown key 'Nu llabilityOfRet'}}
