diff --git a/clang-tools-extra/clang-doc/CMakeLists.txt b/clang-tools-extra/clang-doc/CMakeLists.txt
deleted file mode 100644
index 975ad8e242e4..000000000000
--- a/clang-tools-extra/clang-doc/CMakeLists.txt
+++ /dev/null
@@ -1,37 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  support
-  BitstreamReader
-  FrontendOpenMP
-  )
-
-add_clang_library(clangDoc
-  BitcodeReader.cpp
-  BitcodeWriter.cpp
-  ClangDoc.cpp
-  Generators.cpp
-  HTMLGenerator.cpp
-  Mapper.cpp
-  MDGenerator.cpp
-  Representation.cpp
-  Serialize.cpp
-  YAMLGenerator.cpp
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-)
-
-clang_target_link_libraries(clangDoc
-  PRIVATE
-  clangAnalysis
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangFrontend
-  clangIndex
-  clangLex
-  clangTooling
-  clangToolingCore
-  )
-
-add_subdirectory(tool)
