diff --git a/clang-tools-extra/clangd/test/initialize-params-invalid.test b/clang-tools-extra/clangd/test/initialize-params-invalid.test
deleted file mode 100644
index f51a7279ca27..000000000000
--- a/clang-tools-extra/clangd/test/initialize-params-invalid.test
+++ /dev/null
@@ -1,12 +0,0 @@
-# RUN: clangd -lit-test < %s | FileCheck -strict-whitespace %s
-# Test with invalid initialize request parameters
-{"jsonrpc":"2.0","id":0,"method":"initialize","params":{"processId":"","rootUri":"test:///workspace","capabilities":{},"trace":"off"}}
-#      CHECK:  "id": 0,
-# CHECK-NEXT:  "jsonrpc": "2.0",
-# CHECK-NEXT:  "result": {
-# CHECK-NEXT:    "capabilities": {
-# ...
----
-{"jsonrpc":"2.0","id":3,"method":"shutdown"}
----
-{"jsonrpc":"2.0","method":"exit"}
