diff --git a/clang-tools-extra/clangd/tool/ClangdMain.h b/clang-tools-extra/clangd/tool/ClangdMain.h
deleted file mode 100644
index bd0f51aac83b..000000000000
--- a/clang-tools-extra/clangd/tool/ClangdMain.h
+++ /dev/null
@@ -1,19 +0,0 @@
-//===--- ClangdMain.h - clangd main function ------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_TOOLS_EXTRA_CLANGD_TOOL_CLANGDMAIN_H
-#define LLVM_CLANG_TOOLS_EXTRA_CLANGD_TOOL_CLANGDMAIN_H
-
-namespace clang {
-namespace clangd {
-// clangd main function (clangd server loop)
-int clangdMain(int argc, char *argv[]);
-} // namespace clangd
-} // namespace clang
-
-#endif // LLVM_CLANG_TOOLS_EXTRA_CLANGD_TOOL_CLANGDMAIN_H
