diff --git a/clang/test/APINotes/instancetype.m b/clang/test/APINotes/instancetype.m
deleted file mode 100644
index e3c13188ae9f..000000000000
--- a/clang/test/APINotes/instancetype.m
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: rm -rf %t
-// RUN: %clang_cc1 -fmodules -fimplicit-module-maps -fmodules-cache-path=%t/ModulesCache -fapinotes-modules -fsyntax-only -I %S/Inputs/Headers -verify %s
-
-@import InstancetypeModule;
-
-void test() {
-  // The nullability is here to verify that the API notes were applied.
-  int good = [SomeSubclass instancetypeFactoryMethod]; // expected-error {{initializing 'int' with an expression of type 'SomeSubclass * _Nonnull'}}
-  int bad = [SomeSubclass staticFactoryMethod]; // expected-error {{initializing 'int' with an expression of type 'SomeBaseClass * _Nonnull'}}
-}
