diff --git a/clang-tools-extra/test/clang-tidy/checkers/altera/Inputs/kernel-name-restriction/vhdl_number_two.cl b/clang-tools-extra/test/clang-tidy/checkers/altera/Inputs/kernel-name-restriction/vhdl_number_two.cl
deleted file mode 100644
index c196407d4b5f..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/altera/Inputs/kernel-name-restriction/vhdl_number_two.cl
+++ /dev/null
@@ -1 +0,0 @@
-const int VHDLNUMBERTWOINT = 3;
