diff --git a/clang/lib/Index/CMakeLists.txt b/clang/lib/Index/CMakeLists.txt
deleted file mode 100644
index b4e294304f11..000000000000
--- a/clang/lib/Index/CMakeLists.txt
+++ /dev/null
@@ -1,32 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Core
-  Support
-  )
-
-add_clang_library(clangIndex
-  CommentToXML.cpp
-  FileIndexRecord.cpp
-  IndexBody.cpp
-  IndexDecl.cpp
-  IndexingAction.cpp
-  IndexingContext.cpp
-  IndexSymbol.cpp
-  IndexTypeSourceInfo.cpp
-  USRGeneration.cpp
-
-  ADDITIONAL_HEADERS
-  IndexingContext.h
-
-  LINK_LIBS
-  clangAST
-  clangBasic
-  clangFormat
-  clangFrontend
-  clangLex
-  clangSerialization
-  clangToolingCore
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
