diff --git a/clang-tools-extra/test/clang-apply-replacements/Inputs/crlf/crlf.cpp b/clang-tools-extra/test/clang-apply-replacements/Inputs/crlf/crlf.cpp
deleted file mode 100644
index 26f79968f556..000000000000
--- a/clang-tools-extra/test/clang-apply-replacements/Inputs/crlf/crlf.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-
-// This file intentionally uses a CRLF newlines!
-
-void foo() {
-  int *x = 0;
-}
