diff --git a/clang/examples/PluginsOrder/CMakeLists.txt b/clang/examples/PluginsOrder/CMakeLists.txt
deleted file mode 100644
index c22b0081ad68..000000000000
--- a/clang/examples/PluginsOrder/CMakeLists.txt
+++ /dev/null
@@ -1,11 +0,0 @@
-add_llvm_library(PluginsOrder MODULE PluginsOrder.cpp PLUGIN_TOOL clang)
-
-if(WIN32 OR CYGWIN)
-  set(LLVM_LINK_COMPONENTS
-    Support
-  )
-  clang_target_link_libraries(PluginsOrder PRIVATE
-    clangAST
-    clangFrontend
-    )
-endif()
