diff --git a/clang-tools-extra/test/modularize/Inputs/HeaderGuard.h b/clang-tools-extra/test/modularize/Inputs/HeaderGuard.h
deleted file mode 100644
index 64d747693c06..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/HeaderGuard.h
+++ /dev/null
@@ -1,6 +0,0 @@
-#ifndef _HEADERGUARD_H_
-#define _HEADERGUARD_H_
-#include "HeaderGuardSub1.h"
-#include "HeaderGuardSub2.h"
-#include "HeaderGuardSubSub.h"
-#endif // _HEADERGUARD_H_
