diff --git a/clang/bindings/python/clang/__init__.py b/clang/bindings/python/clang/__init__.py
deleted file mode 100644
index 8d0d1729e4a1..000000000000
--- a/clang/bindings/python/clang/__init__.py
+++ /dev/null
@@ -1,22 +0,0 @@
-# ===- __init__.py - Clang Python Bindings --------------------*- python -*--===#
-#
-# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-# See https://llvm.org/LICENSE.txt for license information.
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-#
-# ===------------------------------------------------------------------------===#
-
-r"""
-Clang Library Bindings
-======================
-
-This package provides access to the Clang compiler and libraries.
-
-The available modules are:
-
-  cindex
-
-    Bindings for the Clang indexing library.
-"""
-
-__all__ = ["cindex"]
