diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/make-unique-cxx11.cpp b/clang-tools-extra/test/clang-tidy/checkers/modernize/make-unique-cxx11.cpp
deleted file mode 100644
index e2944b8080c5..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/make-unique-cxx11.cpp
+++ /dev/null
@@ -1,9 +0,0 @@
-// RUN: %check_clang_tidy -std=c++11 %s modernize-make-unique %t -- -- -I %S/Inputs/smart-ptr
-
-#include "unique_ptr.h"
-// CHECK-FIXES: #include "unique_ptr.h"
-
-void f() {
-  auto my_ptr = std::unique_ptr<int>(new int(1));
-  // CHECK-FIXES: auto my_ptr = std::unique_ptr<int>(new int(1));
-}
