diff --git a/clang/include/clang/Driver/DriverDiagnostic.h b/clang/include/clang/Driver/DriverDiagnostic.h
deleted file mode 100644
index ec2f8b403b0b..000000000000
--- a/clang/include/clang/Driver/DriverDiagnostic.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- DiagnosticDriver.h - Diagnostics for libdriver ---------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_DRIVER_DRIVERDIAGNOSTIC_H
-#define LLVM_CLANG_DRIVER_DRIVERDIAGNOSTIC_H
-
-#include "clang/Basic/DiagnosticDriver.h"
-
-#endif
