diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/3/.clang-tidy b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/3/.clang-tidy
deleted file mode 100644
index 9365108255bd..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/3/.clang-tidy
+++ /dev/null
@@ -1,4 +0,0 @@
-InheritParentConfig: true
-Checks: 'from-child3'
-HeaderFilterRegex: 'child3'
-ExcludeHeaderFilterRegex: 'exc-child3'
