diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/use-using/modernize-use-using.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/use-using/modernize-use-using.h
deleted file mode 100644
index c82be21fcf1e..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/use-using/modernize-use-using.h
+++ /dev/null
@@ -1,6 +0,0 @@
-#ifndef MODERNIZE_USE_USING_H
-#define MODERNIZE_USE_USING_H
-
-typedef int mytype;
-
-#endif
