diff --git a/clang/include/clang/Serialization/SerializationDiagnostic.h b/clang/include/clang/Serialization/SerializationDiagnostic.h
deleted file mode 100644
index 7fc93c11932f..000000000000
--- a/clang/include/clang/Serialization/SerializationDiagnostic.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- SerializationDiagnostic.h - Serialization Diagnostics -*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_SERIALIZATION_SERIALIZATIONDIAGNOSTIC_H
-#define LLVM_CLANG_SERIALIZATION_SERIALIZATIONDIAGNOSTIC_H
-
-#include "clang/Basic/DiagnosticSerialization.h"
-
-#endif
