diff --git a/clang/include/clang/CrossTU/CrossTUDiagnostic.h b/clang/include/clang/CrossTU/CrossTUDiagnostic.h
deleted file mode 100644
index 95a648a56082..000000000000
--- a/clang/include/clang/CrossTU/CrossTUDiagnostic.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- CrossTUDiagnostic.h - Diagnostics for Cross TU ---------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_CROSSTU_CROSSTUDIAGNOSTIC_H
-#define LLVM_CLANG_CROSSTU_CROSSTUDIAGNOSTIC_H
-
-#include "clang/Basic/DiagnosticCrossTU.h"
-
-#endif // LLVM_CLANG_CROSSTU_CROSSTUDIAGNOSTIC_H
