diff --git a/clang-tools-extra/pseudo/test/Unit/lit.site.cfg.py.in b/clang-tools-extra/pseudo/test/Unit/lit.site.cfg.py.in
deleted file mode 100644
index 4107b0d2eb83..000000000000
--- a/clang-tools-extra/pseudo/test/Unit/lit.site.cfg.py.in
+++ /dev/null
@@ -1,11 +0,0 @@
-@LIT_SITE_CFG_IN_HEADER@
-# This is a shim to run the gtest unittests in ../unittests using lit.
-
-config.llvm_libs_dir = "@LLVM_LIBS_DIR@"
-config.shlibdir = "@SHLIBDIR@"
-
-config.clang_pseudo_source_dir = "@CMAKE_CURRENT_SOURCE_DIR@/.."
-config.clang_pseudo_binary_dir = "@CMAKE_CURRENT_BINARY_DIR@/.."
-
-# Delegate logic to lit.cfg.py.
-lit_config.load_config(config, "@CMAKE_CURRENT_SOURCE_DIR@/Unit/lit.cfg.py")
