diff --git a/clang-tools-extra/test/clang-move/Inputs/macro_helper_test.cpp b/clang-tools-extra/test/clang-move/Inputs/macro_helper_test.cpp
deleted file mode 100644
index c2c1e1e1aae6..000000000000
--- a/clang-tools-extra/test/clang-move/Inputs/macro_helper_test.cpp
+++ /dev/null
@@ -1,13 +0,0 @@
-#include "macro_helper_test.h"
-
-#define DEFINE(name) \
-  namespace ns { \
-  static const bool t1 = false; \
-  bool t2_##name = t1; \
-  bool t3_##name = t1; \
-  } \
-  using ns::t2_##name;
-
-DEFINE(test)
-
-void f1() {}
