diff --git a/clang/examples/Attribute/CMakeLists.txt b/clang/examples/Attribute/CMakeLists.txt
deleted file mode 100644
index 770fa28364b7..000000000000
--- a/clang/examples/Attribute/CMakeLists.txt
+++ /dev/null
@@ -1,11 +0,0 @@
-add_llvm_library(Attribute MODULE Attribute.cpp PLUGIN_TOOL clang)
-
-if(WIN32 OR CYGWIN)
-  target_link_libraries(Attribute PRIVATE
-    clangAST
-    clangBasic
-    clangFrontend
-    clangLex
-    LLVMSupport
-    )
-endif()
