diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/.clang-tidy b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/.clang-tidy
deleted file mode 100644
index 74556fa4be84..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/.clang-tidy
+++ /dev/null
@@ -1,5 +0,0 @@
-Checks: '-*,modernize-loop-convert,modernize-use-using'
-CheckOptions:
-  modernize-loop-convert.MaxCopySize: '10'
-  modernize-loop-convert.MinConfidence: reasonable
-  modernize-use-using.IgnoreMacros: true
