diff --git a/clang-tools-extra/clang-tidy/llvmlibc/NamespaceConstants.h b/clang-tools-extra/clang-tidy/llvmlibc/NamespaceConstants.h
deleted file mode 100644
index 83908a7875d0..000000000000
--- a/clang-tools-extra/clang-tidy/llvmlibc/NamespaceConstants.h
+++ /dev/null
@@ -1,20 +0,0 @@
-//===--- NamespaceConstants.h -----------------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#include "llvm/ADT/StringRef.h"
-
-namespace clang::tidy::llvm_libc {
-
-const static llvm::StringRef RequiredNamespaceRefStart = "__llvm_libc";
-const static llvm::StringRef RequiredNamespaceRefMacroName = "LIBC_NAMESPACE";
-const static llvm::StringRef RequiredNamespaceDeclStart =
-    "[[gnu::visibility(\"hidden\")]] __llvm_libc";
-const static llvm::StringRef RequiredNamespaceDeclMacroName =
-    "LIBC_NAMESPACE_DECL";
-
-} // namespace clang::tidy::llvm_libc
