diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor.h b/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor.h
deleted file mode 100644
index dfe5f9733afb..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor.h
+++ /dev/null
@@ -1,5 +0,0 @@
-#ifndef FOO
-#ifndef FOO // this would warn, but not in a header
-void f();
-#endif
-#endif
