diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header.h
deleted file mode 100644
index c2103cb3fc7a..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header.h
+++ /dev/null
@@ -1,10 +0,0 @@
-class ThreadId {
-public:
-  ThreadId(const ThreadId &) {}
-  ThreadId(ThreadId &&) {}
-};
-
-struct A {
-  A(const ThreadId &tid) : threadid(tid) {}
-  ThreadId threadid;
-};
