diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-single-line.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-single-line.cpp
deleted file mode 100644
index 3b90299bd986..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/braces-around-statements-single-line.cpp
+++ /dev/null
@@ -1,33 +0,0 @@
-// RUN: %check_clang_tidy %s readability-braces-around-statements %t -- -config="{CheckOptions: {readability-braces-around-statements.ShortStatementLines: 2}}" --
-
-void do_something(const char *) {}
-
-bool cond(const char *) {
-  return false;
-}
-
-void test() {
-  if (cond("if1") /*comment*/) do_something("same-line");
-
-  if (cond("if2"))
-    do_something("single-line");
-
-  if (cond("if3") /*comment*/)
-    // some comment
-    do_something("three"
-                 "lines");
-  // CHECK-MESSAGES: :[[@LINE-4]]:31: warning: statement should be inside braces
-  // CHECK-FIXES: if (cond("if3") /*comment*/) {
-  // CHECK-FIXES: }
-
-  if (cond("if4") /*comment*/)
-    // some comment
-    do_something("many"
-                 "many"
-                 "many"
-                 "many"
-                 "lines");
-  // CHECK-MESSAGES: :[[@LINE-7]]:31: warning: statement should be inside braces
-  // CHECK-FIXES: if (cond("if4") /*comment*/) {
-  // CHECK-FIXES: }
-}
