diff --git a/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/Modules/module.private.modulemap b/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/Modules/module.private.modulemap
deleted file mode 100644
index 0958a14d6710..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/Modules/module.private.modulemap
+++ /dev/null
@@ -1,5 +0,0 @@
-framework module TopLevelPrivateKit_Private {
-  umbrella header "TopLevelPrivateKit_Private.h"
-  export *
-  module * { export * }
-}
