diff --git a/clang/lib/Analysis/FlowSensitive/Models/CMakeLists.txt b/clang/lib/Analysis/FlowSensitive/Models/CMakeLists.txt
deleted file mode 100644
index 89bbe8791eb2..000000000000
--- a/clang/lib/Analysis/FlowSensitive/Models/CMakeLists.txt
+++ /dev/null
@@ -1,11 +0,0 @@
-add_clang_library(clangAnalysisFlowSensitiveModels
-  ChromiumCheckModel.cpp
-  UncheckedOptionalAccessModel.cpp
-
-  LINK_LIBS
-  clangAnalysis
-  clangAnalysisFlowSensitive
-  clangAST
-  clangASTMatchers
-  clangBasic
-  )
