diff --git a/clang-tools-extra/test/clang-reorder-fields/ClassDifferentFieldsAccesses.cpp b/clang-tools-extra/test/clang-reorder-fields/ClassDifferentFieldsAccesses.cpp
deleted file mode 100644
index dd0a555a815a..000000000000
--- a/clang-tools-extra/test/clang-reorder-fields/ClassDifferentFieldsAccesses.cpp
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: clang-reorder-fields -record-name Foo -fields-order z,y,x %s -- | FileCheck %s
-
-// The order of fields should not change.
-class Foo {
-public:
-  int x;  // CHECK:       {{^  int x;}}
-
-private:
-  int y;  // CHECK:       {{^  int y;}}
-  int z;  // CHECK-NEXT:  {{^  int z;}}
-};
-
-int main() {
-  Foo foo;
-  return 0;
-}
