diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-alias-decls.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/unused-alias-decls.cpp
deleted file mode 100644
index 6d63df4047a4..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-alias-decls.cpp
+++ /dev/null
@@ -1,12 +0,0 @@
-// RUN: %check_clang_tidy %s misc-unused-alias-decls %t
-
-namespace my_namespace {
-class C {};
-}
-
-namespace unused_alias = ::my_namespace; // eol-comments aren't removed (yet)
-// CHECK-MESSAGES: :[[@LINE-1]]:11: warning: namespace alias decl 'unused_alias' is unused
-// CHECK-FIXES: {{^}}// eol-comments aren't removed (yet)
-
-namespace used_alias = ::my_namespace;
-void f() { used_alias::C c; }
