diff --git a/clang/include/clang/Serialization/CMakeLists.txt b/clang/include/clang/Serialization/CMakeLists.txt
deleted file mode 100644
index d91513da9997..000000000000
--- a/clang/include/clang/Serialization/CMakeLists.txt
+++ /dev/null
@@ -1,9 +0,0 @@
-clang_tablegen(AttrPCHRead.inc -gen-clang-attr-pch-read
-  -I ${CMAKE_CURRENT_SOURCE_DIR}/../../
-  SOURCE ../Basic/Attr.td
-  TARGET ClangAttrPCHRead)
-
-clang_tablegen(AttrPCHWrite.inc -gen-clang-attr-pch-write
-  -I ${CMAKE_CURRENT_SOURCE_DIR}/../../
-  SOURCE ../Basic/Attr.td
-  TARGET ClangAttrPCHWrite)
