diff --git a/clang-tools-extra/test/clang-tidy/checkers/cert/limited-randomness.cpp b/clang-tools-extra/test/clang-tidy/checkers/cert/limited-randomness.cpp
deleted file mode 100644
index 845b7350de9f..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/cert/limited-randomness.cpp
+++ /dev/null
@@ -1,28 +0,0 @@
-// RUN: %check_clang_tidy %s cert-msc50-cpp %t
-
-int rand();
-int rand(int);
-
-namespace std {
-using ::rand;
-}
-
-namespace nonstd {
-  int rand();
-}
-
-void testFunction1() {
-  int i = std::rand();
-  // CHECK-MESSAGES: :[[@LINE-1]]:11: warning: rand() has limited randomness; use C++11 random library instead [cert-msc50-cpp]
-
-  int j = ::rand();
-  // CHECK-MESSAGES: :[[@LINE-1]]:11: warning: rand() has limited randomness; use C++11 random library instead [cert-msc50-cpp]
-
-  int k = rand(i);
-
-  int l = nonstd::rand();
-
-  int m = rand();
-  // CHECK-MESSAGES: :[[@LINE-1]]:11: warning: rand() has limited randomness; use C++11 random library instead [cert-msc50-cpp]
-}
-
