diff --git a/clang-tools-extra/test/modularize/Inputs/TemplateClasses.h b/clang-tools-extra/test/modularize/Inputs/TemplateClasses.h
deleted file mode 100644
index 3eab6d900910..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/TemplateClasses.h
+++ /dev/null
@@ -1,15 +0,0 @@
-// Exercise some template issues.  Should not produce errors.
-
-// Forward declaration.
-template<class T> class TemplateClass;
-
-// Full declaration.
-template<class T>class TemplateClass {
-public:
-  TemplateClass() {}
-private:
-  T Member;
-};
-
-// Template alias.
-template<class T> using TemplateClassAlias = TemplateClass<T>;
