diff --git a/clang-tools-extra/clang-include-fixer/find-all-symbols/CMakeLists.txt b/clang-tools-extra/clang-include-fixer/find-all-symbols/CMakeLists.txt
deleted file mode 100644
index 0ed46428b0d6..000000000000
--- a/clang-tools-extra/clang-include-fixer/find-all-symbols/CMakeLists.txt
+++ /dev/null
@@ -1,31 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  FrontendOpenMP
-  )
-
-add_clang_library(findAllSymbols
-  FindAllSymbols.cpp
-  FindAllSymbolsAction.cpp
-  FindAllMacros.cpp
-  HeaderMapCollector.cpp
-  PathConfig.cpp
-  PragmaCommentHandler.cpp
-  STLPostfixHeaderMap.cpp
-  SymbolInfo.cpp
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
-
-clang_target_link_libraries(findAllSymbols
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangFrontend
-  clangLex
-  clangTooling
-  )
-
-add_subdirectory(tool)
