diff --git a/clang/.clang-tidy b/clang/.clang-tidy
deleted file mode 100644
index ba55beb095f5..000000000000
--- a/clang/.clang-tidy
+++ /dev/null
@@ -1,5 +0,0 @@
-# Note that the readability-identifier-naming check is disabled, there are too
-# many violations in the codebase and they create too much noise in clang-tidy
-# results.
-Checks: '-readability-identifier-naming'
-InheritParentConfig: true
