diff --git a/clang-tools-extra/test/clang-reorder-fields/CStructFieldsOrder.cpp b/clang-tools-extra/test/clang-reorder-fields/CStructFieldsOrder.cpp
deleted file mode 100644
index 2ed3578df656..000000000000
--- a/clang-tools-extra/test/clang-reorder-fields/CStructFieldsOrder.cpp
+++ /dev/null
@@ -1,16 +0,0 @@
-// RUN: clang-reorder-fields -record-name ::bar::Foo -fields-order z,w,y,x %s -- | FileCheck %s
-
-namespace bar {
-struct Foo {
-  const int* x; // CHECK:      {{^  double z;}}
-  int y;        // CHECK-NEXT: {{^  int w;}}
-  double z;     // CHECK-NEXT: {{^  int y;}}
-  int w;        // CHECK-NEXT: {{^  const int\* x}}
-};
-} // end namespace bar
-
-int main() {
-  const int x = 13;
-  bar::Foo foo = { &x, 0, 1.29, 17 }; // CHECK: {{^  bar::Foo foo = { 1.29, 17, 0, &x };}} 
-  return 0;
-}
