diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header-with-fix.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header-with-fix.h
deleted file mode 100644
index 62609e25f682..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/pass-by-value/header-with-fix.h
+++ /dev/null
@@ -1,8 +0,0 @@
-struct S {
-  S(S&&);
-  S(const S&);
-};
-struct Foo {
-  Foo(const S &s);
-  S s;
-};
