diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/inconsistent-declaration-parameter-name-strict.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/inconsistent-declaration-parameter-name-strict.cpp
deleted file mode 100644
index f5d86a2bed1c..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/inconsistent-declaration-parameter-name-strict.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: %check_clang_tidy %s readability-inconsistent-declaration-parameter-name %t -- \
-// RUN:   -config="{CheckOptions: {readability-inconsistent-declaration-parameter-name.Strict: true}}"
-
-void inconsistentFunction(int a, int b, int c);
-// CHECK-MESSAGES: :[[@LINE-1]]:6: warning: function 'inconsistentFunction' has 1 other declaration with different parameter names
-void inconsistentFunction(int prefixA, int b, int cSuffix);
-// CHECK-MESSAGES: :[[@LINE-1]]:6: note: the 1st inconsistent declaration seen here
-// CHECK-MESSAGES: :[[@LINE-2]]:6: note: differing parameters are named here: ('prefixA', 'cSuffix'), in the other declaration: ('a', 'c')
-void inconsistentFunction(int a, int b, int c);
-void inconsistentFunction(int /*c*/, int /*c*/, int /*c*/);
