diff --git a/clang-tools-extra/test/clang-tidy/checkers/bugprone/bad-signal-to-kill-thread-undef-sigterm.cpp b/clang-tools-extra/test/clang-tidy/checkers/bugprone/bad-signal-to-kill-thread-undef-sigterm.cpp
deleted file mode 100644
index 7c2ae16a08e8..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/bugprone/bad-signal-to-kill-thread-undef-sigterm.cpp
+++ /dev/null
@@ -1,11 +0,0 @@
-// RUN: clang-tidy %s --checks=-*,bugprone-bad-signal-to-kill-thread -- | count 0
-
-#define SIGTERM 15
-#undef SIGTERM // no-crash
-using pthread_t = int;
-int pthread_kill(pthread_t thread, int sig);
-
-int func() {
-  pthread_t thread;
-  return pthread_kill(thread, 0);
-}
