diff --git a/clang/include/clang/Basic/Version.inc.in b/clang/include/clang/Basic/Version.inc.in
deleted file mode 100644
index 370da8506d14..000000000000
--- a/clang/include/clang/Basic/Version.inc.in
+++ /dev/null
@@ -1,6 +0,0 @@
-#define CLANG_VERSION @CLANG_VERSION@
-#define CLANG_VERSION_STRING "@CLANG_VERSION@"
-#define CLANG_VERSION_MAJOR @CLANG_VERSION_MAJOR@
-#define CLANG_VERSION_MAJOR_STRING "@CLANG_VERSION_MAJOR@"
-#define CLANG_VERSION_MINOR @CLANG_VERSION_MINOR@
-#define CLANG_VERSION_PATCHLEVEL @CLANG_VERSION_PATCHLEVEL@
