diff --git a/clang/lib/Tooling/Syntax/CMakeLists.txt b/clang/lib/Tooling/Syntax/CMakeLists.txt
deleted file mode 100644
index 6588eba2cd37..000000000000
--- a/clang/lib/Tooling/Syntax/CMakeLists.txt
+++ /dev/null
@@ -1,22 +0,0 @@
-set(LLVM_LINK_COMPONENTS Support)
-
-add_clang_library(clangToolingSyntax
-  BuildTree.cpp
-  ComputeReplacements.cpp
-  Nodes.cpp
-  Mutations.cpp
-  TokenBufferTokenManager.cpp
-  Synthesis.cpp
-  Tokens.cpp
-  Tree.cpp
-
-  LINK_LIBS
-  clangAST
-  clangBasic
-  clangFrontend
-  clangLex
-  clangToolingCore
-
-  DEPENDS
-  omp_gen
-  )
