diff --git a/clang-tools-extra/clang-tidy/llvm/CMakeLists.txt b/clang-tools-extra/clang-tidy/llvm/CMakeLists.txt
deleted file mode 100644
index b56498bdc8c4..000000000000
--- a/clang-tools-extra/clang-tidy/llvm/CMakeLists.txt
+++ /dev/null
@@ -1,31 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  FrontendOpenMP
-  Support
-  )
-
-add_clang_library(clangTidyLLVMModule
-  HeaderGuardCheck.cpp
-  IncludeOrderCheck.cpp
-  LLVMTidyModule.cpp
-  PreferIsaOrDynCastInConditionalsCheck.cpp
-  PreferRegisterOverUnsignedCheck.cpp
-  TwineLocalCheck.cpp
-
-  LINK_LIBS
-  clangTidy
-  clangTidyReadabilityModule
-  clangTidyUtils
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
-
-clang_target_link_libraries(clangTidyLLVMModule
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangLex
-  clangTooling
-  )
