diff --git a/clang-tools-extra/test/modularize/Inputs/CoverageNoProblems/Includes1/.hidden/DontFindMe.h b/clang-tools-extra/test/modularize/Inputs/CoverageNoProblems/Includes1/.hidden/DontFindMe.h
deleted file mode 100644
index f60b66cd1f62..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/CoverageNoProblems/Includes1/.hidden/DontFindMe.h
+++ /dev/null
@@ -1,3 +0,0 @@
-#error DontFindMe.h shouldn't be found.
-
-
