diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/dump-config-invalid.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/dump-config-invalid.cpp
deleted file mode 100644
index 4737b49bffe4..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/dump-config-invalid.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: clang-tidy --checks="-*,modernize-make-shared" \
-// RUN:   --config="CheckOptions: { \
-// RUN:      modernize-make-shared.IncludeStyle: '0' }" \
-// RUN:   --dump-config -- | FileCheck %s
-
-// CHECK: modernize-make-shared.IncludeStyle: llvm
