diff --git a/clang/include/clang/Driver/aarch64-mlr-for-calls-only.c b/clang/include/clang/Driver/aarch64-mlr-for-calls-only.c
deleted file mode 100644
index e71a4cdfe0e7..000000000000
--- a/clang/include/clang/Driver/aarch64-mlr-for-calls-only.c
+++ /dev/null
@@ -1,3 +0,0 @@
-// RUN: %clang --target=aarch64-none-gnu -mlr-for-calls-only -### %s 2> %t
-// RUN: FileCheck --check-prefix=CHECK < %t %s
-// CHECK: "-target-feature" "+reserve-lr-for-ra"
