diff --git a/clang/test/APINotes/Inputs/APINotes/SomeOtherKit.apinotes b/clang/test/APINotes/Inputs/APINotes/SomeOtherKit.apinotes
deleted file mode 100644
index ccdc4e15d34d..000000000000
--- a/clang/test/APINotes/Inputs/APINotes/SomeOtherKit.apinotes
+++ /dev/null
@@ -1,8 +0,0 @@
-Name: SomeOtherKit
-Classes:
-  - Name: A
-    Methods:
-      - Selector:        "methodB"
-        MethodKind:      Instance
-        Availability:    none
-        AvailabilityMsg: "anything but this"
