diff --git a/clang-tools-extra/clang-apply-replacements/tool/CMakeLists.txt b/clang-tools-extra/clang-apply-replacements/tool/CMakeLists.txt
deleted file mode 100644
index 1ed734c11433..000000000000
--- a/clang-tools-extra/clang-apply-replacements/tool/CMakeLists.txt
+++ /dev/null
@@ -1,19 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
-
-add_clang_tool(clang-apply-replacements
-  ClangApplyReplacementsMain.cpp
-  )
-clang_target_link_libraries(clang-apply-replacements
-  PRIVATE
-  clangBasic
-  clangFormat
-  clangRewrite
-  clangToolingCore
-  clangToolingRefactoring
-  )
-target_link_libraries(clang-apply-replacements
-  PRIVATE
-  clangApplyReplacements
-  )
