diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-noexcept-error.cpp b/clang-tools-extra/test/clang-tidy/checkers/modernize/use-noexcept-error.cpp
deleted file mode 100644
index 9a80b075d65e..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-noexcept-error.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: %check_clang_tidy -expect-clang-tidy-error %s modernize-use-noexcept %t
-
-// We're not interested in the check issuing a warning here, just making sure
-// clang-tidy doesn't assert.
-undefined_type doesThrow() throw();
-// CHECK-MESSAGES: :[[@LINE-1]]:1: error: unknown type name 'undefined_type' [clang-diagnostic-error]
