diff --git a/clang-tools-extra/test/modularize/ProblemsCompileError.modularize b/clang-tools-extra/test/modularize/ProblemsCompileError.modularize
deleted file mode 100644
index a7ad29862b16..000000000000
--- a/clang-tools-extra/test/modularize/ProblemsCompileError.modularize
+++ /dev/null
@@ -1,3 +0,0 @@
-# RUN: not modularize %S/Inputs/CompileError/module.modulemap 2>&1 | FileCheck %s
-
-# CHECK: {{.*}}{{[/\\]}}Inputs{{[/\\]}}CompileError{{[/\\]}}HasError.h:1:9: error: unknown type name 'WithoutDep'
