diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor-ifdef.cpp b/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor-ifdef.cpp
deleted file mode 100644
index 72b608b1c2c0..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/redundant-preprocessor-ifdef.cpp
+++ /dev/null
@@ -1,36 +0,0 @@
-// RUN: %check_clang_tidy %s readability-redundant-preprocessor %t -- -- -DFOO
-
-// Positive testing.
-#ifdef FOO
-// CHECK-NOTES: [[@LINE+1]]:2: warning: nested redundant #ifdef; consider removing it [readability-redundant-preprocessor]
-#ifdef FOO
-// CHECK-NOTES: [[@LINE-3]]:2: note: previous #ifdef was here
-void f();
-#endif
-#endif
-
-// Positive testing of inverted condition.
-#ifdef FOO
-// CHECK-NOTES: [[@LINE+1]]:2: warning: nested redundant #ifndef; consider removing it [readability-redundant-preprocessor]
-#ifndef FOO
-// CHECK-NOTES: [[@LINE-3]]:2: note: previous #ifdef was here
-void f2();
-#endif
-#endif
-
-// Negative testing.
-#ifdef BAR
-void g();
-#endif
-
-#ifdef FOO
-#ifdef BAR
-void h();
-#endif
-#endif
-
-#ifdef FOO
-#ifndef BAR
-void i();
-#endif
-#endif
