diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/bitint-no-crash.c b/clang-tools-extra/test/clang-tidy/checkers/readability/bitint-no-crash.c
deleted file mode 100644
index f8660924cbef..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/bitint-no-crash.c
+++ /dev/null
@@ -1,6 +0,0 @@
-// RUN: %check_clang_tidy %s readability-magic-numbers %t --
-
-// Don't crash
-
-_BitInt(128) A = 4533629751480627964421wb;
-// CHECK-MESSAGES: warning
