diff --git a/clang-tools-extra/test/clang-tidy/checkers/performance/Inputs/unnecessary-value-param/header-fixed.h b/clang-tools-extra/test/clang-tidy/checkers/performance/Inputs/unnecessary-value-param/header-fixed.h
deleted file mode 100644
index a40b2b2ece52..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/performance/Inputs/unnecessary-value-param/header-fixed.h
+++ /dev/null
@@ -1,15 +0,0 @@
-// struct ABC is expensive to copy and should be
-// passed as a const reference.
-struct ABC {
-  ABC(const ABC&);
-  int get(int) const;
-};
-
-
-int f1(int n,              const ABC& v1,   const ABC& v2); // line 9
-
-int f1(int n, ABC v1); // line 11
-
-
-
-int f2(        int n,       const ABC& v2); // line 15
