diff --git a/clang-tools-extra/test/modularize/Inputs/InconsistentHeader1.h b/clang-tools-extra/test/modularize/Inputs/InconsistentHeader1.h
deleted file mode 100644
index 7dfcca948a18..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/InconsistentHeader1.h
+++ /dev/null
@@ -1,4 +0,0 @@
-// Define symbol such that a declaration exists when this header
-// is included, but not when InconsistentHeader2.h is included.
-#define SYMBOL1 1
-#include "InconsistentSubHeader.h"
