diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-equals-default-cxx98.cpp b/clang-tools-extra/test/clang-tidy/checkers/modernize/use-equals-default-cxx98.cpp
deleted file mode 100644
index d43c1e58c260..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/use-equals-default-cxx98.cpp
+++ /dev/null
@@ -1,8 +0,0 @@
-// RUN: %check_clang_tidy -std=c++98 %s modernize-use-equals-default %t
-
-struct S {
-  S() {}
-  // CHECK-FIXES: S() {}
-  ~S() {}
-  // CHECK-FIXES: ~S() {}
-};
