diff --git a/clang-tools-extra/test/clang-apply-replacements/Inputs/format/no.cpp b/clang-tools-extra/test/clang-apply-replacements/Inputs/format/no.cpp
deleted file mode 100644
index 5bc90811dfb6..000000000000
--- a/clang-tools-extra/test/clang-apply-replacements/Inputs/format/no.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-class C {};
-
-void f() { // This comment necessary to prevent formatting as void f() { ... }
-  C *a = new C();
-  // CHECK: {{^\ \ auto\ a\ \=\ new\ C\(\);}}
-}
