diff --git a/clang-tools-extra/clangd/test/Inputs/background-index/compile_commands.json.tmpl b/clang-tools-extra/clangd/test/Inputs/background-index/compile_commands.json.tmpl
deleted file mode 100644
index 1bb835f642da..000000000000
--- a/clang-tools-extra/clangd/test/Inputs/background-index/compile_commands.json.tmpl
+++ /dev/null
@@ -1,5 +0,0 @@
-[{
-  "directory": "DIRECTORY",
-  "command": "clang foo.cpp",
-  "file": "DIRECTORY/foo.cpp"
-}]
