diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-using-decls-errors.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/unused-using-decls-errors.cpp
deleted file mode 100644
index 1c26a6961aad..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/unused-using-decls-errors.cpp
+++ /dev/null
@@ -1,12 +0,0 @@
-// RUN: %check_clang_tidy -expect-clang-tidy-error %s misc-unused-using-decls %t
-
-namespace n {
-class C;
-}
-
-using n::C;
-
-void f() {
-  for (C *p : unknown()) {}
-  // CHECK-MESSAGES: :[[@LINE-1]]:15: error: use of undeclared identifier 'unknown' [clang-diagnostic-error]
-}
