diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/unique_ptr.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/unique_ptr.h
deleted file mode 100644
index 5dc9e02b637a..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/unique_ptr.h
+++ /dev/null
@@ -1,28 +0,0 @@
-namespace std {
-
-template <typename T>
-class default_delete {};
-
-template <typename type, typename Deleter = std::default_delete<type>>
-class unique_ptr {
-public:
-  unique_ptr() {}
-  unique_ptr(type *ptr) {}
-  unique_ptr(const unique_ptr<type> &t) = delete;
-  unique_ptr(unique_ptr<type> &&t) {}
-  ~unique_ptr() {}
-  type &operator*() { return *ptr; }
-  type *operator->() { return ptr; }
-  type *release() { return ptr; }
-  void reset() {}
-  void reset(type *pt) {}
-  void reset(type pt) {}
-  unique_ptr &operator=(unique_ptr &&) { return *this; }
-  template <typename T>
-  unique_ptr &operator=(unique_ptr<T> &&) { return *this; }
-
-private:
-  type *ptr;
-};
-
-}  // namespace std
