diff --git a/clang/test/APINotes/Inputs/Headers/Methods.apinotes b/clang/test/APINotes/Inputs/Headers/Methods.apinotes
deleted file mode 100644
index 0fa6991a51ff..000000000000
--- a/clang/test/APINotes/Inputs/Headers/Methods.apinotes
+++ /dev/null
@@ -1,8 +0,0 @@
----
-Name: Methods
-Tags:
-- Name: IntWrapper
-  Methods:
-  - Name: getIncremented
-    Availability: none
-    AvailabilityMsg: "oh no"
