diff --git a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/system-other.h b/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/system-other.h
deleted file mode 100644
index 28c26b73ba6b..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/system-other.h
+++ /dev/null
@@ -1,16 +0,0 @@
-//===--- system-other.h - Stub header for tests -----------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef _SYSTEM_OTHER_H_
-#define _SYSTEM_OTHER_H_
-
-// Special system calls.
-
-void other_call();
-
-#endif // _SYSTEM_OTHER_H_
