diff --git a/clang/docs/analyzer/checkers/unix_malloc_example.c b/clang/docs/analyzer/checkers/unix_malloc_example.c
deleted file mode 100644
index 68c5a4a8f1c5..000000000000
--- a/clang/docs/analyzer/checkers/unix_malloc_example.c
+++ /dev/null
@@ -1,30 +0,0 @@
-
-void test() {
-  int *p = malloc(1);
-  free(p);
-  free(p); // warn: attempt to free released memory
-}
-
-void test() {
-  int *p = malloc(sizeof(int));
-  free(p);
-  *p = 1; // warn: use after free
-}
-
-void test() {
-  int *p = malloc(1);
-  if (p)
-    return; // warn: memory is never released
-}
-
-void test() {
-  int a[] = { 1 };
-  free(a); // warn: argument is not allocated by malloc
-}
-
-void test() {
-  int *p = malloc(sizeof(char));
-  p = p - 1;
-  free(p); // warn: argument to free() is offset by -4 bytes
-}
-
