diff --git a/clang-tools-extra/test/modularize/Inputs/NamespaceClasses.h b/clang-tools-extra/test/modularize/Inputs/NamespaceClasses.h
deleted file mode 100644
index 57e6091fd961..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/NamespaceClasses.h
+++ /dev/null
@@ -1,20 +0,0 @@
-// Define same class name in different namespaces.
-
-namespace Namespace1 {
-  class NamespaceClass {
-  public:
-    NamespaceClass() : Member(0) {}
-  private:
-    int Member;
-  };
-}
-
-namespace Namespace2 {
-  class NamespaceClass {
-  public:
-    NamespaceClass() : Member(0) {}
-  private:
-    int Member;
-  };
-}
-
