diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/2nd-translation-unit.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/2nd-translation-unit.cpp
deleted file mode 100644
index 0497037d31b8..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/2nd-translation-unit.cpp
+++ /dev/null
@@ -1,6 +0,0 @@
-
-class A { A(int i); };
-
-// NOLINTBEGIN
-class B { B(int i); };
-// NOLINTEND
