diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-cxx17.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-cxx17.cpp
deleted file mode 100644
index d6e9ebe49a97..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/unconventional-assign-operator-cxx17.cpp
+++ /dev/null
@@ -1,12 +0,0 @@
-// RUN: %check_clang_tidy -std=c++14-or-later %s misc-unconventional-assign-operator %t -- -- -fno-delayed-template-parsing
-
-struct BadModifier {
-  BadModifier& operator=(const BadModifier&) const;
-  // CHECK-MESSAGES: :[[@LINE-1]]:3: warning: operator=() should not be marked 'const'
-};
-
-struct PR35468 {
-  template<typename T> auto &operator=(const T &) {
-    return *this;
-  }
-};
