diff --git a/clang/test/APINotes/Inputs/Frameworks/LayeredKit.framework/Modules/module.modulemap b/clang/test/APINotes/Inputs/Frameworks/LayeredKit.framework/Modules/module.modulemap
deleted file mode 100644
index 04bbe72a2b6e..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/LayeredKit.framework/Modules/module.modulemap
+++ /dev/null
@@ -1,5 +0,0 @@
-framework module LayeredKit {
-  umbrella header "LayeredKit.h"
-  export *
-  module * { export * }
-}
