diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/use-internal-linkage-fix-mode-none.cpp b/clang-tools-extra/test/clang-tidy/checkers/misc/use-internal-linkage-fix-mode-none.cpp
deleted file mode 100644
index 3f2f5897bf71..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/use-internal-linkage-fix-mode-none.cpp
+++ /dev/null
@@ -1,10 +0,0 @@
-// RUN: %check_clang_tidy %s misc-use-internal-linkage %t -- \
-// RUN:   -config="{CheckOptions: {misc-use-internal-linkage.FixMode: 'None'}}"  -- -I%S/Inputs/use-internal-linkage
-
-void func() {}
-// CHECK-MESSAGES: :[[@LINE-1]]:6: warning: function 'func'
-// CHECK-FIXES-NOT: static void func() {}
-
-int global;
-// CHECK-MESSAGES: :[[@LINE-1]]:5: warning: variable 'global'
-// CHECK-FIXES-NOT: static int global;
