diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/shared_ptr.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/shared_ptr.h
deleted file mode 100644
index 337cb28228b0..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/smart-ptr/shared_ptr.h
+++ /dev/null
@@ -1,33 +0,0 @@
-namespace std {
-
-template <typename type>
-class __shared_ptr {
-protected:
-  __shared_ptr();
-  __shared_ptr(type *ptr);
-  ~__shared_ptr();
-public:
-  type &operator*() { return *ptr; }
-  type *operator->() { return ptr; }
-  type *release();
-  void reset();
-  void reset(type *pt);
-
-private:
-  type *ptr;
-};
-
-template <typename type>
-class shared_ptr : public __shared_ptr<type> {
-public:
-  shared_ptr();
-  shared_ptr(type *ptr);
-  shared_ptr(const shared_ptr<type> &t);
-  shared_ptr(shared_ptr<type> &&t);
-  ~shared_ptr();
-  shared_ptr &operator=(shared_ptr &&);
-  template <typename T>
-  shared_ptr &operator=(shared_ptr<T> &&);
-};
-
-}  // namespace std
