diff --git a/clang-tools-extra/include-cleaner/test/Unit/lit.site.cfg.py.in b/clang-tools-extra/include-cleaner/test/Unit/lit.site.cfg.py.in
deleted file mode 100644
index 4926a1a3288b..000000000000
--- a/clang-tools-extra/include-cleaner/test/Unit/lit.site.cfg.py.in
+++ /dev/null
@@ -1,10 +0,0 @@
-@LIT_SITE_CFG_IN_HEADER@
-# This is a shim to run the gtest unittests in ../unittests using lit.
-
-config.llvm_libs_dir = path("@LLVM_LIBS_DIR@")
-config.shlibdir = path("@SHLIBDIR@")
-
-config.clang_include_cleaner_binary_dir = path("@CMAKE_CURRENT_BINARY_DIR@/..")
-
-# Delegate logic to lit.cfg.py.
-lit_config.load_config(config, "@CMAKE_CURRENT_SOURCE_DIR@/Unit/lit.cfg.py")
