diff --git a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstddef b/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstddef
deleted file mode 100644
index 800285e887cd..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/Inputs/Headers/cstddef
+++ /dev/null
@@ -1,19 +0,0 @@
-//===--- cstddef - Stub header for tests -------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef _CSTDDEF_
-#define _CSTDDEF_
-
-#include "stddef.h"
-
-namespace std {
-  using ::ptrdiff_t;
-  using ::size_t;
-}
-
-#endif // _CSTDDEF_
