diff --git a/clang-tools-extra/clang-tidy/boost/CMakeLists.txt b/clang-tools-extra/clang-tidy/boost/CMakeLists.txt
deleted file mode 100644
index fed3c3ba01c1..000000000000
--- a/clang-tools-extra/clang-tidy/boost/CMakeLists.txt
+++ /dev/null
@@ -1,26 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  support
-  FrontendOpenMP
-  )
-
-add_clang_library(clangTidyBoostModule
-  BoostTidyModule.cpp
-  UseRangesCheck.cpp
-  UseToStringCheck.cpp
-
-  LINK_LIBS
-  clangTidy
-  clangTidyUtils
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
-
-clang_target_link_libraries(clangTidyBoostModule
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangLex
-  )
