diff --git a/clang/test/APINotes/Inputs/Headers/InstancetypeModule.apinotes b/clang/test/APINotes/Inputs/Headers/InstancetypeModule.apinotes
deleted file mode 100644
index 813eb506f39a..000000000000
--- a/clang/test/APINotes/Inputs/Headers/InstancetypeModule.apinotes
+++ /dev/null
@@ -1,10 +0,0 @@
-Name: InstancetypeModule
-Classes:
-- Name: SomeBaseClass
-  Methods:
-  - Selector: instancetypeFactoryMethod
-    MethodKind: Class
-    ResultType: SomeBaseClass * _Nonnull
-  - Selector: staticFactoryMethod
-    MethodKind: Class
-    ResultType: SomeBaseClass * _Nonnull
