diff --git a/clang-tools-extra/clang-query/CMakeLists.txt b/clang-tools-extra/clang-query/CMakeLists.txt
deleted file mode 100644
index 34f018c4a03f..000000000000
--- a/clang-tools-extra/clang-query/CMakeLists.txt
+++ /dev/null
@@ -1,26 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  lineeditor
-  support
-  FrontendOpenMP
-  )
-
-add_clang_library(clangQuery
-  Query.cpp
-  QueryParser.cpp
-
-  DEPENDS
-  omp_gen
-  ClangDriverOptions
-  )
-
-clang_target_link_libraries(clangQuery
-  PRIVATE
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangDynamicASTMatchers
-  clangFrontend
-  clangSerialization
-  )
-
-add_subdirectory(tool)
