diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/static-analyzer-config.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/static-analyzer-config.cpp
deleted file mode 100644
index 725f87717346..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/static-analyzer-config.cpp
+++ /dev/null
@@ -1,20 +0,0 @@
-// REQUIRES: static-analyzer
-// RUN: clang-tidy %s -checks='-*,clang-analyzer-unix.Malloc' -config='{CheckOptions: { "clang-analyzer-unix.DynamicMemoryModeling:Optimistic": true}}' -- | FileCheck %s
-typedef __typeof(sizeof(int)) size_t;
-void *malloc(size_t);
-void free(void *);
-void __attribute((ownership_returns(malloc))) *my_malloc(size_t);
-void __attribute((ownership_takes(malloc, 1))) my_free(void *);
-
-void f1() {
-  void *p = malloc(12);
-  return;
-  // CHECK: warning: Potential leak of memory pointed to by 'p' [clang-analyzer-unix.Malloc]
-}
-
-void af2() {
-  void *p = my_malloc(12);
-  my_free(p);
-  free(p);
-  // CHECK: warning: Attempt to free released memory [clang-analyzer-unix.Malloc]
-}
