diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/nolint_in_include.inc b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/nolint_in_include.inc
deleted file mode 100644
index 7c3dfefe247a..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/nolint_in_include.inc
+++ /dev/null
@@ -1,3 +0,0 @@
-// NOLINTBEGIN
-class H { H(int i); };
-// NOLINTEND
