diff --git a/clang-tools-extra/test/clang-move/Inputs/test.cpp b/clang-tools-extra/test/clang-move/Inputs/test.cpp
deleted file mode 100644
index fa777f01b958..000000000000
--- a/clang-tools-extra/test/clang-move/Inputs/test.cpp
+++ /dev/null
@@ -1,11 +0,0 @@
-#include "test.h"
-#include "test2.h"
-
-namespace a {
-int Foo::f() {
-  return 0;
-}
-int Foo::f2(int a, int b) {
-  return a + b;
-}
-} // namespace a
