diff --git a/clang/examples/CMakeLists.txt b/clang/examples/CMakeLists.txt
deleted file mode 100644
index 2396ecac16b2..000000000000
--- a/clang/examples/CMakeLists.txt
+++ /dev/null
@@ -1,13 +0,0 @@
-if(NOT CLANG_BUILD_EXAMPLES)
-  set_property(DIRECTORY PROPERTY EXCLUDE_FROM_ALL ON)
-  set(EXCLUDE_FROM_ALL ON)
-endif()
-
-if(CLANG_PLUGIN_SUPPORT)
-  add_subdirectory(LLVMPrintFunctionNames)
-  add_subdirectory(PrintFunctionNames)
-  add_subdirectory(AnnotateFunctions)
-  add_subdirectory(Attribute)
-  add_subdirectory(CallSuperAttribute)
-  add_subdirectory(PluginsOrder)
-endif()
