diff --git a/clang/lib/ExtractAPI/CMakeLists.txt b/clang/lib/ExtractAPI/CMakeLists.txt
deleted file mode 100644
index f508ffc862cb..000000000000
--- a/clang/lib/ExtractAPI/CMakeLists.txt
+++ /dev/null
@@ -1,21 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  TargetParser
-  )
-
-add_clang_library(clangExtractAPI
-  API.cpp
-  APIIgnoresList.cpp
-  ExtractAPIConsumer.cpp
-  DeclarationFragments.cpp
-  Serialization/SymbolGraphSerializer.cpp
-  TypedefUnderlyingTypeResolver.cpp
-
-  LINK_LIBS
-  clangAST
-  clangBasic
-  clangFrontend
-  clangIndex
-  clangInstallAPI
-  clangLex
-  )
