diff --git a/clang/lib/Headers/__stdarg_header_macro.h b/clang/lib/Headers/__stdarg_header_macro.h
deleted file mode 100644
index beb92ee02526..000000000000
--- a/clang/lib/Headers/__stdarg_header_macro.h
+++ /dev/null
@@ -1,12 +0,0 @@
-/*===---- __stdarg_header_macro.h ------------------------------------------===
- *
- * Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
- * See https://llvm.org/LICENSE.txt for license information.
- * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
- *
- *===-----------------------------------------------------------------------===
- */
-
-#ifndef __STDARG_H
-#define __STDARG_H
-#endif
