diff --git a/clang/lib/Tooling/Transformer/CMakeLists.txt b/clang/lib/Tooling/Transformer/CMakeLists.txt
deleted file mode 100644
index 4ab4bd6f0898..000000000000
--- a/clang/lib/Tooling/Transformer/CMakeLists.txt
+++ /dev/null
@@ -1,25 +0,0 @@
-set(LLVM_LINK_COMPONENTS
-  FrontendOpenMP
-  Support
-)
-
-add_clang_library(clangTransformer
-  Parsing.cpp
-  RangeSelector.cpp
-  RewriteRule.cpp
-  SourceCode.cpp
-  SourceCodeBuilders.cpp
-  Stencil.cpp
-  Transformer.cpp
-
-  LINK_LIBS
-  clangAST
-  clangASTMatchers
-  clangBasic
-  clangLex
-  clangToolingCore
-  clangToolingRefactoring
-
-  DEPENDS
-  omp_gen
-  )
