diff --git a/clang/include/clang/Tooling/Refactoring/RefactoringDiagnostic.h b/clang/include/clang/Tooling/Refactoring/RefactoringDiagnostic.h
deleted file mode 100644
index 967e7b5860ff..000000000000
--- a/clang/include/clang/Tooling/Refactoring/RefactoringDiagnostic.h
+++ /dev/null
@@ -1,14 +0,0 @@
-//===--- RefactoringDiagnostic.h - ------------------------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef LLVM_CLANG_TOOLING_REFACTORING_REFACTORINGDIAGNOSTIC_H
-#define LLVM_CLANG_TOOLING_REFACTORING_REFACTORINGDIAGNOSTIC_H
-
-#include "clang/Basic/DiagnosticRefactoring.h"
-
-#endif // LLVM_CLANG_TOOLING_REFACTORING_REFACTORINGDIAGNOSTIC_H
