diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/global-style1/header.h b/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/global-style1/header.h
deleted file mode 100644
index abbf7dfa4839..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/global-style1/header.h
+++ /dev/null
@@ -1,7 +0,0 @@
-
-
-void style_first_good();
-
-void styleFirstBad();
-
-int thisIsMainLikeIgnored(int argc, const char *argv[]) {}
