diff --git a/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/user-header.h b/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/user-header.h
deleted file mode 100644
index f03982ce73b4..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/readability/Inputs/identifier-naming/user-header.h
+++ /dev/null
@@ -1,17 +0,0 @@
-#ifndef USER_HEADER_H
-#define USER_HEADER_H
-
-#define USER_MACRO(m) int m = 0
-
-namespace USER_NS {
-class object {
-  int member;
-};
-
-float global;
-
-void function() {
-}
-}
-
-#endif // USER_HEADER_H
