diff --git a/clang/test/APINotes/Inputs/Frameworks/SomeOtherKit.framework/Modules/module.modulemap b/clang/test/APINotes/Inputs/Frameworks/SomeOtherKit.framework/Modules/module.modulemap
deleted file mode 100644
index 0aaad92e041c..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/SomeOtherKit.framework/Modules/module.modulemap
+++ /dev/null
@@ -1,5 +0,0 @@
-framework module SomeOtherKit {
-  umbrella header "SomeOtherKit.h"
-  export *
-  module * { export * }
-}
