diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/1st-translation-unit.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/1st-translation-unit.cpp
deleted file mode 100644
index d90fc069c978..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/nolintbeginend/1st-translation-unit.cpp
+++ /dev/null
@@ -1,5 +0,0 @@
-// NOLINTBEGIN
-class A { A(int i); };
-// NOLINTEND
-
-class B { B(int i); };
