diff --git a/clang-tools-extra/pseudo/test/cxx/dangling-else.cpp b/clang-tools-extra/pseudo/test/cxx/dangling-else.cpp
deleted file mode 100644
index 151f3931b53f..000000000000
--- a/clang-tools-extra/pseudo/test/cxx/dangling-else.cpp
+++ /dev/null
@@ -1,22 +0,0 @@
-// RUN: clang-pseudo -grammar=cxx -source=%s --start-symbol=statement-seq --print-forest | FileCheck %s
-
-// Verify the else should belong to the nested if statement
-if (true) if (true) {} else {}
-
-// CHECK:      statement-seq~selection-statement := IF ( condition ) statement
-// CHECK-NEXT: ├─IF
-// CHECK-NEXT: ├─(
-// CHECK-NEXT: ├─condition~TRUE
-// CHECK-NEXT: ├─)
-// CHECK-NEXT: └─statement~selection-statement
-// CHECK-NEXT:   ├─IF
-// CHECK-NEXT:   ├─(
-// CHECK-NEXT:   ├─condition~TRUE
-// CHECK-NEXT:   ├─)
-// CHECK-NEXT:   ├─statement~compound-statement := { }
-// CHECK-NEXT:   │ ├─{
-// CHECK-NEXT:   │ └─}
-// CHECK-NEXT:   ├─ELSE
-// CHECK-NEXT:   └─statement~compound-statement := { }
-// CHECK-NEXT:     ├─{
-// CHECK-NEXT:     └─}
