diff --git a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.third-d.hpp b/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.third-d.hpp
deleted file mode 100644
index e9f288a9bace..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/misc/Inputs/header-include-cycle.third-d.hpp
+++ /dev/null
@@ -1,4 +0,0 @@
-#ifndef THIRD
-#define THIRD
-#include "header-include-cycle.fourth-d.hpp"
-#endif
