diff --git a/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/PrivateHeaders/TopLevelPrivateKit_Private.apinotes b/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/PrivateHeaders/TopLevelPrivateKit_Private.apinotes
deleted file mode 100644
index 43323621588b..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/TopLevelPrivateKit.framework/PrivateHeaders/TopLevelPrivateKit_Private.apinotes
+++ /dev/null
@@ -1,4 +0,0 @@
-Name: TopLevelPrivateKit_Private
-Globals:
-- Name: TopLevelPrivateKit_Private
-  Type: float
