diff --git a/clang-tools-extra/test/clang-doc/Inputs/basic-project/src/Calculator.cpp b/clang-tools-extra/test/clang-doc/Inputs/basic-project/src/Calculator.cpp
deleted file mode 100644
index 64f31dbf13d8..000000000000
--- a/clang-tools-extra/test/clang-doc/Inputs/basic-project/src/Calculator.cpp
+++ /dev/null
@@ -1,17 +0,0 @@
-#include "Calculator.h"
-
-int Calculator::add(int a, int b) {
-    return a + b;
-}
-
-int Calculator::subtract(int a, int b) {
-    return a - b;
-}
-
-int Calculator::multiply(int a, int b) {
-    return a * b;
-}
-
-double Calculator::divide(int a, int b) {
-    return static_cast<double>(a) / b;
-}
