diff --git a/clang/test/APINotes/Inputs/Frameworks/Simple.framework/Headers/Simple.h b/clang/test/APINotes/Inputs/Frameworks/Simple.framework/Headers/Simple.h
deleted file mode 100644
index d206a7a36887..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/Simple.framework/Headers/Simple.h
+++ /dev/null
@@ -1,19 +0,0 @@
-#ifndef SIMPLE_H
-#define SIMPLE_H
-
-__attribute__((__objc_root__))
-@interface I
-@property(class, nonatomic, readonly) id nonnullProperty;
-@property(class, nonatomic, readonly) id nonnullNewProperty;
-
-@property(class, nonatomic, readonly) id optionalProperty;
-@property(class, nonatomic, readonly) id optionalNewProperty;
-
-@property(nonatomic, readonly) id unspecifiedProperty;
-@property(nonatomic, readonly) id unspecifiedNewProperty;
-
-@property(nonatomic, readonly) id scalarProperty;
-@property(nonatomic, readonly) id scalarNewProperty;
-@end
-
-#endif
