diff --git a/clang-tools-extra/test/modularize/Inputs/SomeDecls.h b/clang-tools-extra/test/modularize/Inputs/SomeDecls.h
deleted file mode 100644
index 8892b650c181..000000000000
--- a/clang-tools-extra/test/modularize/Inputs/SomeDecls.h
+++ /dev/null
@@ -1,16 +0,0 @@
-// Declare a couple of functions - no modules problems.
-
-void FuncOne();
-
-int FuncTwo(int arg);
-
-void FuncOverload(int arg) {}
-void FuncOverload(char *arg) {}
-
-namespace Namespace1 {
-  void FuncNameSpace() {}
-}
-
-namespace Namespace2 {
-  void FuncNameSpace() {}
-}
