diff --git a/clang/lib/Headers/__stdarg___gnuc_va_list.h b/clang/lib/Headers/__stdarg___gnuc_va_list.h
deleted file mode 100644
index 2a0a7e8cc6c0..000000000000
--- a/clang/lib/Headers/__stdarg___gnuc_va_list.h
+++ /dev/null
@@ -1,13 +0,0 @@
-/*===---- __stdarg___gnuc_va_list.h - Definition of __gnuc_va_list ---------===
- *
- * Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
- * See https://llvm.org/LICENSE.txt for license information.
- * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
- *
- *===-----------------------------------------------------------------------===
- */
-
-#ifndef __GNUC_VA_LIST
-#define __GNUC_VA_LIST
-typedef __builtin_va_list __gnuc_va_list;
-#endif
