diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/key-dict/.clang-tidy b/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/key-dict/.clang-tidy
deleted file mode 100644
index 3abef4360d8d..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/Inputs/config-files/4/key-dict/.clang-tidy
+++ /dev/null
@@ -1,7 +0,0 @@
-InheritParentConfig: true
-Checks: 'llvm-qualified-auto'
-CheckOptions:
-  modernize-loop-convert.MaxCopySize: '20'
-  llvm-qualified-auto.AddConstToQualified: 'true'
-  IgnoreMacros: 'false'
-
