diff --git a/clang/test/APINotes/yaml-reader-errors.m b/clang/test/APINotes/yaml-reader-errors.m
deleted file mode 100644
index 9e5ee34c3e41..000000000000
--- a/clang/test/APINotes/yaml-reader-errors.m
+++ /dev/null
@@ -1,5 +0,0 @@
-// RUN: rm -rf %t
-// RUN: not %clang_cc1 -fmodules -fimplicit-module-maps -fapinotes -fapinotes-modules -fmodules-cache-path=%t -I %S/Inputs/yaml-reader-errors/ -fsyntax-only %s > %t.err 2>&1
-// RUN: FileCheck %S/Inputs/yaml-reader-errors/UIKit.apinotes < %t.err
-
-@import UIKit;
