diff --git a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/concat-nested-namespaces/modernize-concat-nested-namespaces.h b/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/concat-nested-namespaces/modernize-concat-nested-namespaces.h
deleted file mode 100644
index 703d7e05ca2e..000000000000
--- a/clang-tools-extra/test/clang-tidy/checkers/modernize/Inputs/concat-nested-namespaces/modernize-concat-nested-namespaces.h
+++ /dev/null
@@ -1,8 +0,0 @@
-namespace nn1 {
-namespace nn2 {
-// CHECK-FIXES: namespace nn1::nn2
-void t();
-} // namespace nn2
-} // namespace nn1
-// CHECK-FIXES: void t();
-// CHECK-FIXES-NEXT: } // namespace nn1::nn2
