diff --git a/clang-tools-extra/clang-tidy/hicpp/IgnoredRemoveResultCheck.cpp b/clang-tools-extra/clang-tidy/hicpp/IgnoredRemoveResultCheck.cpp
deleted file mode 100644
index b1a18485ce16..000000000000
--- a/clang-tools-extra/clang-tidy/hicpp/IgnoredRemoveResultCheck.cpp
+++ /dev/null
@@ -1,30 +0,0 @@
-//===--- IgnoredRemoveResultCheck.cpp - clang-tidy ------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#include "IgnoredRemoveResultCheck.h"
-
-namespace clang::tidy::hicpp {
-
-IgnoredRemoveResultCheck::IgnoredRemoveResultCheck(llvm::StringRef Name,
-                                                   ClangTidyContext *Context)
-    : UnusedReturnValueCheck(Name, Context,
-                             {
-                                 "::std::remove$",
-                                 "::std::remove_if$",
-                                 "::std::unique$",
-                             }) {
-  // The constructor for ClangTidyCheck needs to have been called
-  // before we can access options via Options.get().
-  AllowCastToVoid = Options.get("AllowCastToVoid", true);
-}
-
-void IgnoredRemoveResultCheck::storeOptions(ClangTidyOptions::OptionMap &Opts) {
-  Options.store(Opts, "AllowCastToVoid", AllowCastToVoid);
-}
-
-} // namespace clang::tidy::hicpp
