diff --git a/clang-tools-extra/test/clang-tidy/infrastructure/empty-database.cpp b/clang-tools-extra/test/clang-tidy/infrastructure/empty-database.cpp
deleted file mode 100644
index 54fe8910a9a9..000000000000
--- a/clang-tools-extra/test/clang-tidy/infrastructure/empty-database.cpp
+++ /dev/null
@@ -1,5 +0,0 @@
-// UNSUPPORTED: system-windows
-
-// RUN: not --crash clang-tidy -p %S/Inputs/empty-database %s 2>&1 | FileCheck %s
-
-// CHECK: LLVM ERROR: Cannot chdir into ""!
