diff --git a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/Headers/SomeKitExplicitNullability.h b/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/Headers/SomeKitExplicitNullability.h
deleted file mode 100644
index 914ab00357fc..000000000000
--- a/clang/test/APINotes/Inputs/Frameworks/SomeKit.framework/Headers/SomeKitExplicitNullability.h
+++ /dev/null
@@ -1,4 +0,0 @@
-@interface A(ExplicitNullabilityProperties)
-@property (nonatomic, readwrite, retain, nonnull) A *explicitNonnullInstance;
-@property (nonatomic, readwrite, retain, nullable) A *explicitNullableInstance;
-@end
