diff --git a/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlType.h b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlType.h
new file mode 100644
index 000000000000..92e4d71ed98d
--- /dev/null
+++ b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlType.h
@@ -0,0 +1,22 @@
+//
+// Created by de29664 on 7/29/24.
+//
+
+#ifndef HAKC_HAKCYAMLTYPE_H
+#define HAKC_HAKCYAMLTYPE_H
+
+#include <string>
+
+#include "llvm/Support/YAMLTraits.h"
+
+namespace llvm::hakc {
+    class HAKCYamlType {
+    public:
+        HAKCYamlType() = default;
+
+        std::string DebugType;
+        std::string LLVMType;
+    };
+} // hakc
+
+#endif //HAKC_HAKCYAMLTYPE_H
