diff --git a/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCPass.h b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCPass.h
new file mode 100644
index 000000000000..6eeb9c02b23e
--- /dev/null
+++ b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCPass.h
@@ -0,0 +1,15 @@
+//
+// Created by derrick on 3/16/21.
+//
+#ifndef PMC_HAKCPASS_H
+#define PMC_HAKCPASS_H
+
+#include "llvm/IR/PassManager.h"
+
+namespace llvm {
+    class HAKCPass : public PassInfoMixin<HAKCPass> {
+    public:
+        PreservedAnalyses run(Module &M, ModuleAnalysisManager &MAM);
+    };
+}
+#endif//PMC_HAKCPASS_H
