diff --git a/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlClique.h b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlClique.h
new file mode 100644
index 000000000000..bd44235c2d52
--- /dev/null
+++ b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCCompartmentalizationPolicy/yaml/HAKCYamlClique.h
@@ -0,0 +1,24 @@
+//
+// Created by de29664 on 7/29/24.
+//
+
+#ifndef HAKC_HAKCYAMLCLIQUE_H
+#define HAKC_HAKCYAMLCLIQUE_H
+
+#include "HAKC-defs.h"
+
+#include "llvm/Support/YAMLTraits.h"
+
+namespace llvm::hakc {
+
+    class HAKCYamlClique {
+    public:
+        HAKCYamlClique() = default;
+
+        hakc_access_token_t AccessToken;
+        hakc_compartment_division_t DivisionID;
+    };
+
+} // hakc
+
+#endif //HAKC_HAKCYAMLCLIQUE_H
