diff --git a/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCTypeIdentifier/HAKCGlobalInfo.h b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCTypeIdentifier/HAKCGlobalInfo.h
new file mode 100644
index 000000000000..b11bc025a023
--- /dev/null
+++ b/llvm/include/llvm/Transforms/Compartmentalization/hakc/HAKCTypeIdentifier/HAKCGlobalInfo.h
@@ -0,0 +1,26 @@
+//
+// Created by de29664 on 5/2/23.
+//
+
+#ifndef HAKC_HAKCGLOBALINFO_H
+#define HAKC_HAKCGLOBALINFO_H
+
+#include "llvm/IR/GlobalVariable.h"
+#include "HAKCSymbolInfo.h"
+
+namespace llvm::hakc {
+
+    class HAKCGlobalInfo : public HAKCSymbolInfo {
+    public:
+        HAKCGlobalInfo(CommonHAKCAnalysis &Analysis, StringRef Name, bool DebugActive);
+
+        void SetGlobalVariable(GlobalVariable *GV);
+
+        GlobalVariable *GetGlobalVariable();
+
+        StringRef GetYamlIdentifier() const override;
+    };
+
+} // hakc
+
+#endif //HAKC_HAKCGLOBALINFO_H
