diff --git a/llvm/lib/Transforms/Compartmentalization/hakc/HAKCFunctionDefinition/HAKCFunctionDefinition.cpp b/llvm/lib/Transforms/Compartmentalization/hakc/HAKCFunctionDefinition/HAKCFunctionDefinition.cpp
new file mode 100644
index 000000000000..1d547af398ea
--- /dev/null
+++ b/llvm/lib/Transforms/Compartmentalization/hakc/HAKCFunctionDefinition/HAKCFunctionDefinition.cpp
@@ -0,0 +1,18 @@
+//
+// Created by de29664 on 6/23/23.
+//
+
+#include "llvm/Transforms/Compartmentalization/hakc/HAKCFunctionDefinition/HAKCFunctionDefinition.h"
+
+namespace llvm::hakc {
+    HAKCFunctionDefinition::HAKCFunctionDefinition(Function *F) : F(F) {
+    }
+
+    StringRef HAKCFunctionDefinition::GetName() const {
+        return F->getName();
+    }
+
+    Function *HAKCFunctionDefinition::GetFunction() const {
+        return F;
+    }
+} // hakc
