diff --git a/clang/include/clang/Basic/Attr.td b/clang/include/clang/Basic/Attr.td
index bfd50f6a6779..f3fc496b0f3c 100644
--- a/clang/include/clang/Basic/Attr.td
+++ b/clang/include/clang/Basic/Attr.td
@@ -3684,3 +3684,17 @@ def EnforceTCBLeaf : InheritableAttr {
   let Documentation = [EnforceTCBLeafDocs];
   bit InheritEvenIfAlreadyPresent = 1;
 }
+
+def KernelUserPtr : InheritableAttr {
+  let Spellings = [GCC<"kernel_user_ptr">];
+  let Subjects = SubjectList<[Named, Value]>;
+  let Documentation = [KernelUserPtrDocs];
+  bit InheritEvenIfAlreadyPresent = 1;
+}
+
+def PerCPUPtr : InheritableAttr {
+  let Spellings = [GCC<"per_cpu_ptr">];
+  let Subjects = SubjectList<[Named, Value]>;
+  let Documentation = [PerCPUPtrDocs];
+  bit InheritEvenIfAlreadyPresent = 1;
+}
diff --git a/clang/include/clang/Basic/AttrDocs.td b/clang/include/clang/Basic/AttrDocs.td
index 170a0fe3d4c4..fd4234105597 100644
--- a/clang/include/clang/Basic/AttrDocs.td
+++ b/clang/include/clang/Basic/AttrDocs.td
@@ -5761,3 +5761,19 @@ def EnforceTCBLeafDocs : Documentation {
   - ``enforce_tcb_leaf(Name)`` indicates that this function is a part of the TCB named ``Name``
   }];
 }
+
+def KernelUserPtrDocs : Documentation {
+  let Category = DocCatVariable;
+  let Heading = "kernel_user_ptr";
+  let Content = [{
+  Adding to ``__user`` in the Linux kernel to signal a pointer is a userspace pointer.
+  }];
+}
+
+def PerCPUPtrDocs : Documentation {
+  let Category = DocCatVariable;
+  let Heading = "per_cpu_ptr";
+  let Content = [{
+  Adding to ``__percpu`` in the Linux kernel to signal a pointer is a percpu pointer.
+  }];
+}
\ No newline at end of file
diff --git a/clang/lib/AST/ASTContext.cpp b/clang/lib/AST/ASTContext.cpp
index cb7f00abf9e9..b02865892be6 100644
--- a/clang/lib/AST/ASTContext.cpp
+++ b/clang/lib/AST/ASTContext.cpp
@@ -10678,6 +10678,11 @@ GVALinkage ASTContext::GetGVALinkageForVariable(const VarDecl *VD) {
 }
 
 bool ASTContext::DeclMustBeEmitted(const Decl *D) {
+  if (D->hasAttr<KernelUserPtrAttr>())
+    return true;
+  if (D->hasAttr<PerCPUPtrAttr>())
+    return true;
+
   if (const auto *VD = dyn_cast<VarDecl>(D)) {
     if (!VD->isFileVarDecl())
       return false;
diff --git a/clang/lib/CodeGen/CGCall.cpp b/clang/lib/CodeGen/CGCall.cpp
index bc7582c67989..2f6e43a7b07e 100644
--- a/clang/lib/CodeGen/CGCall.cpp
+++ b/clang/lib/CodeGen/CGCall.cpp
@@ -1965,6 +1965,10 @@ void CodeGenModule::ConstructAttributeList(
       FuncAttrs.addAttribute(llvm::Attribute::NoDuplicate);
     if (TargetDecl->hasAttr<ConvergentAttr>())
       FuncAttrs.addAttribute(llvm::Attribute::Convergent);
+    if (TargetDecl->hasAttr<KernelUserPtrAttr>())
+      FuncAttrs.addAttribute(llvm::Attribute::KernelUserPtr);
+    if (TargetDecl->hasAttr<PerCPUPtrAttr>())
+      FuncAttrs.addAttribute(llvm::Attribute::PerCPUPtr);
 
     if (const FunctionDecl *Fn = dyn_cast<FunctionDecl>(TargetDecl)) {
       AddAttributesFromFunctionProtoType(
@@ -2438,6 +2442,44 @@ static const NonNullAttr *getNonNullAttr(const Decl *FD, const ParmVarDecl *PVD,
   return nullptr;
 }
 
+static const KernelUserPtrAttr *getKernelUserPtrAttr(const Decl *FD,
+                                                     const ParmVarDecl *PVD,
+                                                     QualType ArgType,
+                                                     unsigned ArgNo) {
+  if (!ArgType->isAnyPointerType() && !ArgType->isBlockPointerType())
+    return nullptr;
+  if (PVD) {
+    if (auto KUPA = PVD->getAttr<KernelUserPtrAttr>())
+      return KUPA;
+  }
+  // Check function attributes.
+  if (!FD)
+    return nullptr;
+  for (const auto *KUPAttr : FD->specific_attrs<KernelUserPtrAttr>()) {
+      return KUPAttr;
+  }
+  return nullptr;
+}
+
+static const PerCPUPtrAttr *getPerCPUPtrAttr(const Decl *FD,
+                                                     const ParmVarDecl *PVD,
+                                                     QualType ArgType,
+                                                     unsigned ArgNo) {
+  if (!ArgType->isAnyPointerType() && !ArgType->isBlockPointerType())
+    return nullptr;
+  if (PVD) {
+    if (auto PCPA = PVD->getAttr<PerCPUPtrAttr>())
+      return PCPA;
+  }
+  // Check function attributes.
+  if (!FD)
+    return nullptr;
+  for (const auto *PCPAttr : FD->specific_attrs<PerCPUPtrAttr>()) {
+    return PCPAttr;
+  }
+  return nullptr;
+}
+
 namespace {
   struct CopyBackSwiftError final : EHScopeStack::Cleanup {
     Address Temp;
@@ -2596,6 +2638,13 @@ void CodeGenFunction::EmitFunctionProlog(const CGFunctionInfo &FI,
               !CGM.getCodeGenOpts().NullPointerIsValid)
             AI->addAttr(llvm::Attribute::NonNull);
 
+          if (getKernelUserPtrAttr(CurCodeDecl, PVD, PVD->getType(),
+                                   PVD->getFunctionScopeIndex()))
+            AI->addAttr(llvm::Attribute::KernelUserPtr);
+          if (getPerCPUPtrAttr(CurCodeDecl, PVD, PVD->getType(),
+                                   PVD->getFunctionScopeIndex()))
+            AI->addAttr(llvm::Attribute::PerCPUPtr);
+
           QualType OTy = PVD->getOriginalType();
           if (const auto *ArrTy =
               getContext().getAsConstantArrayType(OTy)) {
diff --git a/clang/lib/CodeGen/CGDecl.cpp b/clang/lib/CodeGen/CGDecl.cpp
index a01638f0b67b..9888f94a75af 100644
--- a/clang/lib/CodeGen/CGDecl.cpp
+++ b/clang/lib/CodeGen/CGDecl.cpp
@@ -437,6 +437,10 @@ void CodeGenFunction::EmitStaticVarDecl(const VarDecl &D,
     var->addAttribute("rodata-section", SA->getName());
   if (auto *SA = D.getAttr<PragmaClangRelroSectionAttr>())
     var->addAttribute("relro-section", SA->getName());
+  if (auto *KUP = D.getAttr<KernelUserPtrAttr>())
+    var->addAttribute("kernel_user_ptr");
+  if (auto *PCP = D.getAttr<PerCPUPtrAttr>())
+    var->addAttribute("per_cpu_ptr");
 
   if (const SectionAttr *SA = D.getAttr<SectionAttr>())
     var->setSection(SA->getName());
@@ -1608,6 +1612,19 @@ CodeGenFunction::EmitAutoVarAlloca(const VarDecl &D) {
   if (D.hasAttr<AnnotateAttr>() && HaveInsertPoint())
     EmitVarAnnotations(&D, address.getPointer());
 
+  if (D.hasAttr<KernelUserPtrAttr>()) {
+    llvm::Instruction *instruction =
+        dyn_cast<llvm::Instruction>(address.getPointer());
+
+    instruction->addAnnotationMetadata("kernel_user_ptr");
+  }
+  if (D.hasAttr<PerCPUPtrAttr>()) {
+    llvm::Instruction *instruction =
+        dyn_cast<llvm::Instruction>(address.getPointer());
+
+    instruction->addAnnotationMetadata("per_cpu_ptr");
+  }
+
   // Make sure we call @llvm.lifetime.end.
   if (emission.useLifetimeMarkers())
     EHStack.pushCleanup<CallLifetimeEnd>(NormalEHLifetimeMarker,
diff --git a/clang/lib/CodeGen/CGRecordLayoutBuilder.cpp b/clang/lib/CodeGen/CGRecordLayoutBuilder.cpp
index cf8313f92587..7e796c62cbee 100644
--- a/clang/lib/CodeGen/CGRecordLayoutBuilder.cpp
+++ b/clang/lib/CodeGen/CGRecordLayoutBuilder.cpp
@@ -901,6 +901,15 @@ CodeGenTypes::ComputeRecordLayout(const RecordDecl *D, llvm::StructType *Ty) {
   // signifies that the type is no longer opaque and record layout is complete,
   // but we may need to recursively layout D while laying D out as a base type.
   Ty->setBody(Builder.FieldTypes, Builder.Packed);
+  unsigned Idx = 0;
+  for(auto *field : D->fields()) {
+    if(field->hasAttrs()) {
+      Attr *attr = *field->attr_begin();
+      auto lAttr = llvm::Attribute::getAttrKindFromName(attr->getSpelling());
+      Ty->setIndexAttr(Idx, lAttr);
+    }
+    Idx++;
+  }
 
   auto RL = std::make_unique<CGRecordLayout>(
       Ty, BaseTy, (bool)Builder.IsZeroInitializable,
diff --git a/clang/lib/CodeGen/CodeGenModule.cpp b/clang/lib/CodeGen/CodeGenModule.cpp
index 9c9bd4e374af..3dcf2e6579b3 100644
--- a/clang/lib/CodeGen/CodeGenModule.cpp
+++ b/clang/lib/CodeGen/CodeGenModule.cpp
@@ -1912,6 +1912,10 @@ void CodeGenModule::setNonAliasAttributes(GlobalDecl GD,
 
   if (D) {
     if (auto *GV = dyn_cast<llvm::GlobalVariable>(GO)) {
+      if(D->hasAttr<KernelUserPtrAttr>())
+        GV->addAttribute(llvm::Attribute::KernelUserPtr);
+      if(D->hasAttr<PerCPUPtrAttr>())
+        GV->addAttribute(llvm::Attribute::PerCPUPtr);
       if (auto *SA = D->getAttr<PragmaClangBSSSectionAttr>())
         GV->addAttribute("bss-section", SA->getName());
       if (auto *SA = D->getAttr<PragmaClangDataSectionAttr>())
@@ -1923,6 +1927,11 @@ void CodeGenModule::setNonAliasAttributes(GlobalDecl GD,
     }
 
     if (auto *F = dyn_cast<llvm::Function>(GO)) {
+      if(D->hasAttr<KernelUserPtrAttr>())
+        F->addFnAttr(llvm::Attribute::KernelUserPtr);
+      if(D->hasAttr<PerCPUPtrAttr>())
+        F->addFnAttr(llvm::Attribute::PerCPUPtr);
+
       if (auto *SA = D->getAttr<PragmaClangTextSectionAttr>())
         if (!D->getAttr<SectionAttr>())
           F->addFnAttr("implicit-section-name", SA->getName());
@@ -3796,6 +3805,10 @@ CodeGenModule::GetOrCreateLLVMGlobal(StringRef MangledName,
     }
 
     setGVProperties(GV, D);
+    if(D->hasAttr<KernelUserPtrAttr>())
+      GV->addAttribute(llvm::Attribute::KernelUserPtr);
+    if(D->hasAttr<PerCPUPtrAttr>())
+      GV->addAttribute(llvm::Attribute::PerCPUPtr);
 
     // If required by the ABI, treat declarations of static data members with
     // inline initializers as definitions.
diff --git a/clang/lib/Sema/SemaDeclAttr.cpp b/clang/lib/Sema/SemaDeclAttr.cpp
index 30d08b3d4ac0..5940c2f4a4e8 100644
--- a/clang/lib/Sema/SemaDeclAttr.cpp
+++ b/clang/lib/Sema/SemaDeclAttr.cpp
@@ -2239,6 +2239,14 @@ static void handleUnusedAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) UnusedAttr(S.Context, AL));
 }
 
+static void handleKernelUserPtrAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
+  D->addAttr(::new (S.Context) KernelUserPtrAttr(S.Context, AL));
+}
+
+static void handlePerCPUPtrAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
+  D->addAttr(::new (S.Context) PerCPUPtrAttr(S.Context, AL));
+}
+
 static void handleConstructorAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   uint32_t priority = ConstructorAttr::DefaultPriority;
   if (AL.getNumArgs() &&
@@ -8326,6 +8334,14 @@ static void ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D,
   case ParsedAttr::AT_EnforceTCBLeaf:
     handleEnforceTCBAttr<EnforceTCBLeafAttr, EnforceTCBAttr>(S, D, AL);
     break;
+
+  case ParsedAttr::AT_KernelUserPtr:
+    handleKernelUserPtrAttr(S, D, AL);
+    break;
+
+  case ParsedAttr::AT_PerCPUPtr:
+    handlePerCPUPtrAttr(S, D, AL);
+    break;
   }
 }
 
diff --git a/clang/lib/Sema/SemaType.cpp b/clang/lib/Sema/SemaType.cpp
index 4178024d1264..e17aa920c102 100644
--- a/clang/lib/Sema/SemaType.cpp
+++ b/clang/lib/Sema/SemaType.cpp
@@ -8045,6 +8045,14 @@ static void processTypeAttrs(TypeProcessingState &state, QualType &type,
     case ParsedAttr::IgnoredAttribute:
       break;
 
+    case ParsedAttr::AT_KernelUserPtr:
+      attr.setUsedAsTypeAttr();
+      break;
+
+    case ParsedAttr::AT_PerCPUPtr:
+      attr.setUsedAsTypeAttr();
+      break;
+
     case ParsedAttr::AT_MayAlias:
       // FIXME: This attribute needs to actually be handled, but if we ignore
       // it it breaks large amounts of Linux software.
diff --git a/llvm/include/llvm/Bitcode/LLVMBitCodes.h b/llvm/include/llvm/Bitcode/LLVMBitCodes.h
index 5b4854d6c95e..248f832ced55 100644
--- a/llvm/include/llvm/Bitcode/LLVMBitCodes.h
+++ b/llvm/include/llvm/Bitcode/LLVMBitCodes.h
@@ -657,6 +657,8 @@ enum AttributeKindCodes {
   ATTR_KIND_NO_CALLBACK = 71,
   ATTR_KIND_HOT = 72,
   ATTR_KIND_NO_PROFILE = 73,
+  ATTR_KIND_KERNEL_USER_PTR = 74,
+  ATTR_KIND_PER_CPU_PTR = 75,
 };
 
 enum ComdatSelectionKindCodes {
diff --git a/llvm/include/llvm/IR/Attributes.td b/llvm/include/llvm/IR/Attributes.td
index f7ffc888c65a..6ec09308f878 100644
--- a/llvm/include/llvm/IR/Attributes.td
+++ b/llvm/include/llvm/IR/Attributes.td
@@ -259,6 +259,10 @@ def ZExt : EnumAttr<"zeroext">;
 /// Function is required to make Forward Progress.
 def MustProgress : TypeAttr<"mustprogress">;
 
+def KernelUserPtr : TypeAttr<"kernel_user_ptr">;
+
+def PerCPUPtr : TypeAttr<"per_cpu_ptr">;
+
 /// Target-independent string attributes.
 def LessPreciseFPMAD : StrBoolAttr<"less-precise-fpmad">;
 def NoInfsFPMath : StrBoolAttr<"no-infs-fp-math">;
diff --git a/llvm/include/llvm/IR/DerivedTypes.h b/llvm/include/llvm/IR/DerivedTypes.h
index c3d97f4520e1..4d3e206af845 100644
--- a/llvm/include/llvm/IR/DerivedTypes.h
+++ b/llvm/include/llvm/IR/DerivedTypes.h
@@ -21,6 +21,7 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/IR/Type.h"
+#include "llvm/IR/Attributes.h"
 #include "llvm/Support/Casting.h"
 #include "llvm/Support/Compiler.h"
 #include "llvm/Support/TypeSize.h"
@@ -226,6 +227,8 @@ class StructType : public Type {
   /// type that has an empty name.
   void *SymbolTableEntry = nullptr;
 
+  std::map<unsigned, Attribute::AttrKind> MemberAttrs;
+
 public:
   StructType(const StructType &) = delete;
   StructType &operator=(const StructType &) = delete;
@@ -337,6 +340,10 @@ public:
   bool indexValid(const Value *V) const;
   bool indexValid(unsigned Idx) const { return Idx < getNumElements(); }
 
+  /// Methods for querying and setting member attributes
+  bool indexContainsAttr(unsigned Idx, Attribute::AttrKind kind) const;
+  void setIndexAttr(unsigned Idx, Attribute::AttrKind kind);
+
   /// Methods for support type inquiry through isa, cast, and dyn_cast.
   static bool classof(const Type *T) {
     return T->getTypeID() == StructTyID;
diff --git a/llvm/lib/AsmParser/LLLexer.cpp b/llvm/lib/AsmParser/LLLexer.cpp
index 427de74f91ac..8de43bdfdfb7 100644
--- a/llvm/lib/AsmParser/LLLexer.cpp
+++ b/llvm/lib/AsmParser/LLLexer.cpp
@@ -704,6 +704,9 @@ lltok::Kind LLLexer::LexIdentifier() {
   KEYWORD(byref);
   KEYWORD(mustprogress);
 
+  KEYWORD(kernel_user_ptr);
+  KEYWORD(per_cpu_ptr);
+
   KEYWORD(type);
   KEYWORD(opaque);
 
diff --git a/llvm/lib/AsmParser/LLParser.cpp b/llvm/lib/AsmParser/LLParser.cpp
index 2a3fb8fb6658..a8a6e28fd166 100644
--- a/llvm/lib/AsmParser/LLParser.cpp
+++ b/llvm/lib/AsmParser/LLParser.cpp
@@ -1406,6 +1406,10 @@ bool LLParser::parseFnAttributeValuePairs(AttrBuilder &B,
     case lltok::kw_uwtable: B.addAttribute(Attribute::UWTable); break;
     case lltok::kw_willreturn: B.addAttribute(Attribute::WillReturn); break;
     case lltok::kw_writeonly: B.addAttribute(Attribute::WriteOnly); break;
+
+    case lltok::kw_kernel_user_ptr: B.addAttribute(Attribute::KernelUserPtr); break;
+    case lltok::kw_per_cpu_ptr: B.addAttribute(Attribute::PerCPUPtr); break;
+
     case lltok::kw_preallocated: {
       Type *Ty;
       if (parsePreallocated(Ty))
@@ -1764,6 +1768,9 @@ bool LLParser::parseOptionalParamAttrs(AttrBuilder &B) {
     case lltok::kw_zeroext:         B.addAttribute(Attribute::ZExt); break;
     case lltok::kw_immarg:          B.addAttribute(Attribute::ImmArg); break;
 
+    case lltok::kw_per_cpu_ptr:     B.addAttribute(Attribute::PerCPUPtr); break;
+    case lltok::kw_kernel_user_ptr: B.addAttribute(Attribute::KernelUserPtr); break;
+
     case lltok::kw_alignstack:
     case lltok::kw_alwaysinline:
     case lltok::kw_argmemonly:
@@ -1857,6 +1864,9 @@ bool LLParser::parseOptionalReturnAttrs(AttrBuilder &B) {
     case lltok::kw_signext:         B.addAttribute(Attribute::SExt); break;
     case lltok::kw_zeroext:         B.addAttribute(Attribute::ZExt); break;
 
+    case lltok::kw_kernel_user_ptr: B.addAttribute(Attribute::KernelUserPtr); break;
+    case lltok::kw_per_cpu_ptr:     B.addAttribute(Attribute::PerCPUPtr); break;
+
     // error handling.
     case lltok::kw_byval:
     case lltok::kw_inalloca:
diff --git a/llvm/lib/AsmParser/LLToken.h b/llvm/lib/AsmParser/LLToken.h
index 5149f861837a..fdcef3f0e1ff 100644
--- a/llvm/lib/AsmParser/LLToken.h
+++ b/llvm/lib/AsmParser/LLToken.h
@@ -452,6 +452,9 @@ enum Kind {
   kw_bit,
   kw_varFlags,
 
+  kw_kernel_user_ptr,
+  kw_per_cpu_ptr,
+
   // Unsigned Valued tokens (UIntVal).
   LabelID,    // 42:
   GlobalID,   // @42
diff --git a/llvm/lib/Bitcode/Reader/BitcodeReader.cpp b/llvm/lib/Bitcode/Reader/BitcodeReader.cpp
index f2800201e871..83584d0cd1e1 100644
--- a/llvm/lib/Bitcode/Reader/BitcodeReader.cpp
+++ b/llvm/lib/Bitcode/Reader/BitcodeReader.cpp
@@ -1541,6 +1541,10 @@ static Attribute::AttrKind getAttrFromCode(uint64_t Code) {
     return Attribute::MustProgress;
   case bitc::ATTR_KIND_HOT:
     return Attribute::Hot;
+  case bitc::ATTR_KIND_KERNEL_USER_PTR:
+    return Attribute::KernelUserPtr;
+  case bitc::ATTR_KIND_PER_CPU_PTR:
+    return Attribute::PerCPUPtr;
   }
 }
 
diff --git a/llvm/lib/Bitcode/Writer/BitcodeWriter.cpp b/llvm/lib/Bitcode/Writer/BitcodeWriter.cpp
index 37ecb9992e44..11c9348f3116 100644
--- a/llvm/lib/Bitcode/Writer/BitcodeWriter.cpp
+++ b/llvm/lib/Bitcode/Writer/BitcodeWriter.cpp
@@ -754,6 +754,10 @@ static uint64_t getAttrKindEncoding(Attribute::AttrKind Kind) {
     return bitc::ATTR_KIND_BYREF;
   case Attribute::MustProgress:
     return bitc::ATTR_KIND_MUSTPROGRESS;
+  case Attribute::KernelUserPtr:
+    return bitc::ATTR_KIND_KERNEL_USER_PTR;
+  case Attribute::PerCPUPtr:
+    return bitc::ATTR_KIND_PER_CPU_PTR;
   case Attribute::EndAttrKinds:
     llvm_unreachable("Can not encode end-attribute kinds marker.");
   case Attribute::None:
diff --git a/llvm/lib/IR/Attributes.cpp b/llvm/lib/IR/Attributes.cpp
index c4629decc6d9..519a2a6fcc6a 100644
--- a/llvm/lib/IR/Attributes.cpp
+++ b/llvm/lib/IR/Attributes.cpp
@@ -459,6 +459,10 @@ std::string Attribute::getAsString(bool InAttrGrp) const {
     return "noundef";
   if (hasAttribute(Attribute::MustProgress))
     return "mustprogress";
+  if (hasAttribute(Attribute::KernelUserPtr))
+    return "kernel_user_ptr";
+  if (hasAttribute(Attribute::PerCPUPtr))
+    return "per_cpu_ptr";
 
   const bool IsByVal = hasAttribute(Attribute::ByVal);
   if (IsByVal || hasAttribute(Attribute::StructRet)) {
diff --git a/llvm/lib/IR/Type.cpp b/llvm/lib/IR/Type.cpp
index bade7dc325f4..47d4761e5c5a 100644
--- a/llvm/lib/IR/Type.cpp
+++ b/llvm/lib/IR/Type.cpp
@@ -584,6 +584,19 @@ StructType *StructType::getTypeByName(LLVMContext &C, StringRef Name) {
   return C.pImpl->NamedStructTypes.lookup(Name);
 }
 
+bool StructType::indexContainsAttr(unsigned int Idx, Attribute::AttrKind kind) const {
+  if(MemberAttrs.find(Idx) != MemberAttrs.end()) {
+    return MemberAttrs.at(Idx) == kind;
+  }
+  return false;
+}
+
+void StructType::setIndexAttr(unsigned Idx, Attribute::AttrKind kind) {
+  if(indexValid(Idx)) {
+    MemberAttrs[Idx] = kind;
+  }
+}
+
 //===----------------------------------------------------------------------===//
 //                           ArrayType Implementation
 //===----------------------------------------------------------------------===//
diff --git a/llvm/lib/IR/Verifier.cpp b/llvm/lib/IR/Verifier.cpp
index 6dd299ee9845..6eb9fc6f5288 100644
--- a/llvm/lib/IR/Verifier.cpp
+++ b/llvm/lib/IR/Verifier.cpp
@@ -1662,7 +1662,8 @@ static bool isFuncOnlyAttr(Attribute::AttrKind Kind) {
 static bool isFuncOrArgAttr(Attribute::AttrKind Kind) {
   return Kind == Attribute::ReadOnly || Kind == Attribute::WriteOnly ||
          Kind == Attribute::ReadNone || Kind == Attribute::NoFree ||
-         Kind == Attribute::Preallocated;
+         Kind == Attribute::Preallocated || Kind == Attribute::KernelUserPtr ||
+         Kind == Attribute::PerCPUPtr;
 }
 
 void Verifier::verifyAttributeTypes(AttributeSet Attrs, bool IsFunction,
@@ -1718,6 +1719,8 @@ void Verifier::verifyParameterAttrs(AttributeSet Attrs, Type *Ty,
                Attrs.hasAttribute(Attribute::InReg);
   AttrCount += Attrs.hasAttribute(Attribute::Nest);
   AttrCount += Attrs.hasAttribute(Attribute::ByRef);
+  AttrCount += Attrs.hasAttribute(Attribute::KernelUserPtr);
+  AttrCount += Attrs.hasAttribute(Attribute::PerCPUPtr);
   Assert(AttrCount <= 1,
          "Attributes 'byval', 'inalloca', 'preallocated', 'inreg', 'nest', "
          "'byref', and 'sret' are incompatible!",
diff --git a/llvm/lib/Transforms/Utils/AssumeBundleBuilder.cpp b/llvm/lib/Transforms/Utils/AssumeBundleBuilder.cpp
index 3daff3b4430b..37bbc3a96580 100644
--- a/llvm/lib/Transforms/Utils/AssumeBundleBuilder.cpp
+++ b/llvm/lib/Transforms/Utils/AssumeBundleBuilder.cpp
@@ -57,6 +57,8 @@ bool isUsefullToPreserve(Attribute::AttrKind Kind) {
     case Attribute::Dereferenceable:
     case Attribute::DereferenceableOrNull:
     case Attribute::Cold:
+    case Attribute::KernelUserPtr:
+    case Attribute::PerCPUPtr:
       return true;
     default:
       return false;
diff --git a/llvm/lib/Transforms/Utils/CodeExtractor.cpp b/llvm/lib/Transforms/Utils/CodeExtractor.cpp
index 390925a03b73..fe6d5f17e9db 100644
--- a/llvm/lib/Transforms/Utils/CodeExtractor.cpp
+++ b/llvm/lib/Transforms/Utils/CodeExtractor.cpp
@@ -974,6 +974,8 @@ Function *CodeExtractor::constructFunction(const ValueSet &inputs,
       case Attribute::NoCfCheck:
       case Attribute::MustProgress:
       case Attribute::NoProfile:
+      case Attribute::KernelUserPtr:
+      case Attribute::PerCPUPtr:
         break;
       }
 
