diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index 12618966c4ad..4a047ed8ecf0 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -986,6 +986,9 @@ endif()
 
 include(HandleLLVMOptions)
 
+option(LLVM_ENABLE_HAKC "Build HAKC" OFF)
+mark_as_advanced(LLVM_ENABLE_HAKC)
+
 ######
 
 # Configure all of the various header file fragments LLVM uses which depend on
@@ -1423,3 +1426,4 @@ endif()
 if (LLVM_INCLUDE_UTILS AND LLVM_INCLUDE_TOOLS)
   add_subdirectory(utils/llvm-locstats)
 endif()
+
