diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 2d49aea0ff67..55921fc78405 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -9,6 +9,15 @@
 # for more details.
 #
 # Copyright (C) 1995-2001 by Russell King
+# HAKC: check for pass path var once at the very top
+ifeq ($(CONFIG_HAKC), y)
+ifeq ($(CONFIG_HAKC_PASS_PATH),"")
+$(error "HAKC_PASS_PATH is not set!")
+endif
+# the following two lines are common to all supported architectures
+KBUILD_CFLAGS += -fexperimental-new-pass-manager
+KBUILD_CFLAGS += -fpass-plugin=$(CONFIG_HAKC_PASS_PATH)
+endif
 
 LDFLAGS_vmlinux	:=--no-undefined -X
 
@@ -96,6 +105,13 @@ else ifeq ($(CONFIG_AS_HAS_ARMV8_2), y)
   asm-arch := armv8.2-a
 endif
 
+# HAKC: is this still what we want?
+ifeq ($(CONFIG_HAKC_ARM_V9), y)
+asm-arch := armv8.5-a+memtag
+else ifeq ($(CONFIG_HAKC_ARM_V8), y)
+asm-arch := armv8.4-a
+endif
+
 ifdef asm-arch
 KBUILD_CFLAGS	+= -Wa,-march=$(asm-arch) \
 		   -DARM64_ASM_ARCH='"$(asm-arch)"'
