diff --git a/arch/arm64/include/asm/uaccess.h b/arch/arm64/include/asm/uaccess.h
index 14be5000c5a0..779486ec6747 100644
--- a/arch/arm64/include/asm/uaccess.h
+++ b/arch/arm64/include/asm/uaccess.h
@@ -37,6 +37,7 @@ static inline int __access_ok(const void __user *ptr, unsigned long size);
  */
 static inline int access_ok(const void __user *addr, unsigned long size)
 {
+#if !IS_ENABLED(CONFIG_HAKC_ARM_V9)
 	/*
 	 * Asynchronous I/O running in a kernel thread does not have the
 	 * TIF_TAGGED_ADDR flag of the process owning the mm, so always untag
@@ -45,6 +46,7 @@ static inline int access_ok(const void __user *addr, unsigned long size)
 	if (IS_ENABLED(CONFIG_ARM64_TAGGED_ADDR_ABI) &&
 	    (current->flags & PF_KTHREAD || test_thread_flag(TIF_TAGGED_ADDR)))
 		addr = untagged_addr(addr);
+#endif
 
 	return likely(__access_ok(addr, size));
 }
