diff --git a/Documentation/ABI/testing/sysfs-class-led b/Documentation/ABI/testing/sysfs-class-led
index b2ff0012c0f2..a5618d2b5ee4 100644
--- a/Documentation/ABI/testing/sysfs-class-led
+++ b/Documentation/ABI/testing/sysfs-class-led
@@ -63,7 +63,7 @@ What:		/sys/class/leds/<led>/color
 Date:		June 2023
 KernelVersion:	6.5
 Description:
-		Color of the LED.
+		DivisionID of the LED.
 
 		This is a read-only file. Reading this file returns the color
 		of the LED as a string (e.g: "red", "green", "multicolor").
