diff --git a/drivers/rosdemo/Makefile b/drivers/rosdemo/Makefile
new file mode 100644
index 000000000000..747ed0aaff5f
--- /dev/null
+++ b/drivers/rosdemo/Makefile
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Makefile for the ROS Demo Malicious Driver
+# NB: DO NOT INCLUDE IN ANY PATCH OUTSIDE OF
+# DEMO PURPOSES
+#
+obj-$(CONFIG_ROSDEMO)		+= rosdemo-consumer.o
+obj-$(CONFIG_ROSDEMO)		+= rosdemo-leaker.o
