diff --git a/include/linux/sprintf.h b/include/linux/sprintf.h
index 33dcbec71925..c950959b5140 100644
--- a/include/linux/sprintf.h
+++ b/include/linux/sprintf.h
@@ -10,7 +10,13 @@ int num_to_str(char *buf, int size, unsigned long long num, unsigned int width);
 __printf(2, 3) int sprintf(char *buf, const char * fmt, ...);
 __printf(2, 0) int vsprintf(char *buf, const char *, va_list);
 __printf(3, 4) int snprintf(char *buf, size_t size, const char *fmt, ...);
+#ifdef CONFIG_HAKC
+__printf(3, 4) int ls_snprintf(char *buf, size_t size, const char *fmt, ...);
+#endif
 __printf(3, 0) int vsnprintf(char *buf, size_t size, const char *fmt, va_list args);
+#ifdef CONFIG_HAKC
+__printf(3, 0) int ls_vsnprintf(char *buf, size_t size, const char *fmt, va_list args);
+#endif
 __printf(3, 4) int scnprintf(char *buf, size_t size, const char *fmt, ...);
 __printf(3, 0) int vscnprintf(char *buf, size_t size, const char *fmt, va_list args);
 __printf(2, 3) __malloc char *kasprintf(gfp_t gfp, const char *fmt, ...);
