diff --git a/include/asm-generic/vmlinux.lds.h b/include/asm-generic/vmlinux.lds.h
index 67d8dd2f1bde..382f178e86e7 100644
--- a/include/asm-generic/vmlinux.lds.h
+++ b/include/asm-generic/vmlinux.lds.h
@@ -195,6 +195,7 @@
 #define BOUNDED_SECTION_PRE_LABEL(_sec_, _label_, _BEGIN_, _END_)	\
 	_BEGIN_##_label_ = .;						\
 	KEEP(*(_sec_))							\
+	KEEP(*(.hakc.*_CLIQUE##_sec_))				\
 	_END_##_label_ = .;
 
 #define BOUNDED_SECTION_POST_LABEL(_sec_, _label_, _BEGIN_, _END_)	\
@@ -313,6 +314,8 @@
 	. = ALIGN(8);							\
 	__##name##_of_table = .;					\
 	KEEP(*(__##name##_of_table))					\
+	KEEP(*(.hakc.*_CLIQUE__##name##_of_table))			\
+	KEEP(*(.hakc.*_CLIQUE__##name##_of_table_end))			\
 	KEEP(*(__##name##_of_table_end))
 
 #define TIMER_OF_TABLES()	OF_TABLE(CONFIG_TIMER_OF, timer)
@@ -354,6 +357,8 @@
 	*(DATA_MAIN)							\
 	*(.data..decrypted)						\
 	*(.ref.data)							\
+	*(.hakc.*_CLIQUE.ref.data)					\
+	*(.hakc.*_CLIQUE.data)						\
 	*(.data..shared_aligned) /* percpu related */			\
 	MEM_KEEP(init.data*)						\
 	MEM_KEEP(exit.data*)						\
@@ -363,6 +368,7 @@
 	__end_once = .;							\
 	STRUCT_ALIGN();							\
 	*(__tracepoints)						\
+	*(.hakc.*_CLIQUE__tracepoints)					\
 	/* implement dynamic printk debug */				\
 	. = ALIGN(8);							\
 	BOUNDED_SECTION_BY(__dyndbg_classes, ___dyndbg_classes)		\
@@ -391,11 +397,13 @@
 #define READ_MOSTLY_DATA(align)						\
 	. = ALIGN(align);						\
 	*(.data..read_mostly)						\
+	*(.hakc.*_CLIQUE.data..read_mostly)				\
 	. = ALIGN(align);
 
 #define CACHELINE_ALIGNED_DATA(align)					\
 	. = ALIGN(align);						\
-	*(.data..cacheline_aligned)
+	*(.data..cacheline_aligned)					\
+	 *(.hakc.*_CLIQUE.data..cacheline_aligned)
 
 #define INIT_TASK_DATA(align)						\
 	. = ALIGN(align);						\
@@ -429,6 +437,7 @@
 	. = ALIGN(8);							\
 	__start_ro_after_init = .;					\
 	*(.data..ro_after_init)						\
+	*(.hakc.*_CLIQUE.data..ro_after_init)				\
 	JUMP_TABLE_DATA							\
 	STATIC_CALL_DATA						\
 	__end_ro_after_init = .;
@@ -456,11 +465,13 @@
 	.rodata           : AT(ADDR(.rodata) - LOAD_OFFSET) {		\
 		__start_rodata = .;					\
 		*(.rodata) *(.rodata.*)					\
+		*(.hakc.*_CLIQUE.rodata)				\
 		SCHED_DATA						\
 		RO_AFTER_INIT_DATA	/* Read only after init */	\
 		. = ALIGN(8);						\
 		BOUNDED_SECTION_BY(__tracepoints_ptrs, ___tracepoints_ptrs) \
 		*(__tracepoints_strings)/* Tracepoints: strings */	\
+		*(.hakc.*_CLIQUE__tracepoints_strings)			\
 	}								\
 									\
 	.rodata1          : AT(ADDR(.rodata1) - LOAD_OFFSET) {		\
@@ -529,6 +540,12 @@
 		BOUNDED_SECTION_BY(__param, ___param)			\
 	}								\
 									\
+	/* Generated function pointers for module param PAC ctxs. */	\
+	__hakc_modparam_ctx_fp : AT(ADDR(__hakc_modparam_ctx_fp) - LOAD_OFFSET) {	\
+		__start___hakc_modparam_ctx_fp = .;			\
+		KEEP(*(.hakc.modparam_ctx_fp))				\
+		 __stop___hakc_modparam_ctx_fp = .;			\
+        }                                                               \
 	/* Built-in module versions. */					\
 	__modver : AT(ADDR(__modver) - LOAD_OFFSET) {			\
 		BOUNDED_SECTION_BY(__modver, ___modver)			\
@@ -572,6 +589,9 @@
 		*(.text.unknown .text.unknown.*)			\
 		NOINSTR_TEXT						\
 		*(.ref.text)						\
+		*(.hakc.*_CLIQUE.ref.text)				\
+		*(.hakc.*_CLIQUE.text)					\
+		*(.hakc.modparam_ctx.text)				\
 		*(.text.asan.* .text.tsan.*)				\
 	MEM_KEEP(init.text*)						\
 	MEM_KEEP(exit.text*)						\
@@ -583,6 +603,7 @@
 		ALIGN_FUNCTION();					\
 		__sched_text_start = .;					\
 		*(.sched.text)						\
+		*(.hakc.*_CLIQUE.sched.text)				\
 		__sched_text_end = .;
 
 /* spinlock.text is aling to function alignment to secure we have same
@@ -609,6 +630,7 @@
 		ALIGN_FUNCTION();					\
 		__irqentry_text_start = .;				\
 		*(.irqentry.text)					\
+		*(.hakc.*_CLIQUE.irqentry.text)				\
 		__irqentry_text_end = .;
 
 #define SOFTIRQENTRY_TEXT						\
@@ -681,10 +703,12 @@
 #define INIT_DATA							\
 	KEEP(*(SORT(___kentry+*)))					\
 	*(.init.data .init.data.*)					\
+	*(.hakc.*_CLIQUE.init.data)					\
 	MEM_DISCARD(init.data*)						\
 	KERNEL_CTORS()							\
 	MCOUNT_REC()							\
 	*(.init.rodata .init.rodata.*)					\
+	*(.hakc.*_CLIQUE.init.rodata)					\
 	FTRACE_EVENTS()							\
 	TRACE_SYSCALLS()						\
 	KPROBE_BLACKLIST()						\
@@ -708,6 +732,7 @@
 #define INIT_TEXT							\
 	*(.init.text .init.text.*)					\
 	*(.text.startup)						\
+	*(.hakc.*_CLIQUE.init.text)					\
 	MEM_DISCARD(init.text*)
 
 #define EXIT_DATA							\
@@ -720,10 +745,13 @@
 #define EXIT_TEXT							\
 	*(.exit.text)							\
 	*(.text.exit)							\
+	*(.hakc.*_CLIQUE.exit.text)					\
+	*(.hakc.*_CLIQUE.text.exit)					\
 	MEM_DISCARD(exit.text)
 
 #define EXIT_CALL							\
-	*(.exitcall.exit)
+	*(.exitcall.exit)						\
+	*(.hakc.*_CLIQUE.exitcall.exit)
 
 /*
  * bss (Block Started by Symbol) - uninitialized data
@@ -1037,9 +1065,11 @@
 	*(.data..percpu..page_aligned)					\
 	. = ALIGN(cacheline);						\
 	*(.data..percpu..read_mostly)					\
+	*(.hakc.*_CLIQUE.data..percpu..read_mostly)			\
 	. = ALIGN(cacheline);						\
 	*(.data..percpu)						\
 	*(.data..percpu..shared_aligned)				\
+	*(.hakc.*_CLIQUE.data..percpu..shared_aligned)			\
 	PERCPU_DECRYPTED_SECTION					\
 	__per_cpu_end = .;
 
