diff --git a/init/main.c b/init/main.c
index 436d73261810..7d5a365f29d3 100644
--- a/init/main.c
+++ b/init/main.c
@@ -101,6 +101,10 @@
 #include <linux/randomize_kstack.h>
 #include <net/net_namespace.h>
 
+#if IS_ENABLED(CONFIG_HAKC)
+#include <linux/hakc/hakc.h>
+#endif
+
 #include <asm/io.h>
 #include <asm/setup.h>
 #include <asm/sections.h>
@@ -682,6 +686,10 @@ noinline void __ref __noreturn rest_init(void)
 	struct task_struct *tsk;
 	int pid;
 
+#if IS_ENABLED(CONFIG_HAKC_ARM_V8) || IS_ENABLED(CONFIG_HAKC_X86)
+	initialize_hakc();
+#endif
+
 	rcu_scheduler_starting();
 	/*
 	 * We need to spawn init first so that it obtains pid 1, however
