diff --git a/kernel/Makefile b/kernel/Makefile
index 3947122d618b..f06110f1a6de 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -2,7 +2,6 @@
 #
 # Makefile for the linux kernel.
 #
-
 obj-y     = fork.o exec_domain.o panic.o \
 	    cpu.o exit.o softirq.o resource.o \
 	    sysctl.o capability.o ptrace.o user.o \
@@ -12,6 +11,8 @@ obj-y     = fork.o exec_domain.o panic.o \
 	    notifier.o ksysfs.o cred.o reboot.o \
 	    async.o range.o smpboot.o ucount.o regset.o ksyms_common.o
 
+obj-$(CONFIG_HAKC) += hakc/
+
 obj-$(CONFIG_USERMODE_DRIVER) += usermode_driver.o
 obj-$(CONFIG_MULTIUSER) += groups.o
 obj-$(CONFIG_VHOST_TASK) += vhost_task.o
