diff --git a/kernel/module/internal.h b/kernel/module/internal.h
index c8b7b4dcf782..e169357b3a83 100644
--- a/kernel/module/internal.h
+++ b/kernel/module/internal.h
@@ -14,6 +14,10 @@
 #include <linux/rcupdate.h>
 #include <linux/mm.h>
 
+#ifdef CONFIG_HAKC
+#include <linux/hakc/hakc.h>
+#endif
+
 #ifndef ARCH_SHF_SMALL
 #define ARCH_SHF_SMALL 0
 #endif
@@ -81,6 +85,9 @@ struct load_info {
 #endif
 	struct {
 		unsigned int sym, str, mod, vers, info, pcpu;
+#ifdef CONFIG_HAKC
+		unsigned int hakc_pcpu[HAKC_COLOR_COUNT];
+#endif
 	} index;
 };
 
