diff --git a/arch/arm64/kernel/module-plts.c b/arch/arm64/kernel/module-plts.c
index e53493d8b208..a5cc418c8ff3 100644
--- a/arch/arm64/kernel/module-plts.c
+++ b/arch/arm64/kernel/module-plts.c
@@ -310,6 +310,19 @@ int module_frob_arch_sections(Elf_Ehdr *ehdr, Elf_Shdr *sechdrs,
 			tramp = sechdrs + i;
 		else if (sechdrs[i].sh_type == SHT_SYMTAB)
 			syms = (Elf64_Sym *)sechdrs[i].sh_addr;
+
+#if IS_ENABLED(CONFIG_HAKC)
+		if (!mod->hakc_protected &&
+			strstr(secstrings + sechdrs[i].sh_name, ".hakc.")) {
+			mod->hakc_protected = true;
+		}
+		if (mod->hakc_protected) {
+			if (strstr(secstrings + sechdrs[i].sh_name,
+				".data..ro_after_init..data.hakc.")) {
+				sechdrs[i].sh_flags |= SHF_RO_AFTER_INIT;
+			}
+		}
+#endif
 	}
 
 	if (!mod->arch.core.plt_shndx || !mod->arch.init.plt_shndx) {
