diff --git a/arch/arm64/include/asm/uaccess.h b/arch/arm64/include/asm/uaccess.h
index 190b494e22ab..192b6fae3885 100644
--- a/arch/arm64/include/asm/uaccess.h
+++ b/arch/arm64/include/asm/uaccess.h
@@ -37,7 +37,7 @@
 static inline unsigned long __range_ok(const void __user *addr, unsigned long size)
 {
 	unsigned long ret, limit = TASK_SIZE_MAX - 1;
-
+#if !IS_ENABLED(CONFIG_HAKC_ARM_V9)
 	/*
 	 * Asynchronous I/O running in a kernel thread does not have the
 	 * TIF_TAGGED_ADDR flag of the process owning the mm, so always untag
@@ -46,7 +46,7 @@ static inline unsigned long __range_ok(const void __user *addr, unsigned long si
 	if (IS_ENABLED(CONFIG_ARM64_TAGGED_ADDR_ABI) &&
 	    (current->flags & PF_KTHREAD || test_thread_flag(TIF_TAGGED_ADDR)))
 		addr = untagged_addr(addr);
-
+#endif
 	__chk_user_ptr(addr);
 	asm volatile(
 	// A + B <= C + 1 for all A,B,C, in four easy steps:
@@ -179,14 +179,22 @@ static inline void __uaccess_enable_hw_pan(void)
  */
 static inline void __uaccess_disable_tco(void)
 {
+#ifdef CONFIG_HAKC_ARM_V9
+	asm volatile(SET_PSTATE_TCO(0));
+#else
 	asm volatile(ALTERNATIVE("nop", SET_PSTATE_TCO(0),
 				 ARM64_MTE, CONFIG_KASAN_HW_TAGS));
+#endif
 }
 
 static inline void __uaccess_enable_tco(void)
 {
+#ifdef CONFIG_HAKC_ARM_V9
+	asm volatile(SET_PSTATE_TCO(1));
+#else
 	asm volatile(ALTERNATIVE("nop", SET_PSTATE_TCO(1),
 				 ARM64_MTE, CONFIG_KASAN_HW_TAGS));
+#endif
 }
 
 /*
