diff --git a/arch/arm64/kernel/setup.c b/arch/arm64/kernel/setup.c
index be5f85b0a24d..74db28190247 100644
--- a/arch/arm64/kernel/setup.c
+++ b/arch/arm64/kernel/setup.c
@@ -52,6 +52,10 @@
 #include <asm/xen/hypervisor.h>
 #include <asm/mmu_context.h>
 
+#if defined(CONFIG_HAKC_ARM_V9)
+#include <linux/hakc/hakc.h>
+#endif
+
 static int num_standard_resources;
 static struct resource *standard_resources;
 
@@ -349,7 +353,9 @@ void __init __no_sanitize_address setup_arch(char **cmdline_p)
 
 	bootmem_init();
 
+#if !defined(CONFIG_HAKC_ARM_V9)
 	kasan_init();
+#endif
 
 	request_standard_resources();
 
@@ -364,8 +370,12 @@ void __init __no_sanitize_address setup_arch(char **cmdline_p)
 	smp_init_cpus();
 	smp_build_mpidr_hash();
 
+#if defined(CONFIG_HAKC_ARM_V9)
+	hakc_init_tags();
+#else
 	/* Init percpu seeds for random tags after cpus are set up. */
 	kasan_init_sw_tags();
+#endif
 
 #ifdef CONFIG_ARM64_SW_TTBR0_PAN
 	/*
