diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index 2776423a587e..badabf32b631 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -202,8 +202,16 @@ extern __printf(2, 3) int sprintf(char *buf, const char * fmt, ...);
 extern __printf(2, 0) int vsprintf(char *buf, const char *, va_list);
 extern __printf(3, 4)
 int snprintf(char *buf, size_t size, const char *fmt, ...);
+#if defined(CONFIG_HAKC)
+extern __printf(3, 4)
+int ls_snprintf(char *buf, size_t size, const char *fmt, ...);
+#endif
 extern __printf(3, 0)
 int vsnprintf(char *buf, size_t size, const char *fmt, va_list args);
+#if defined(CONFIG_HAKC)
+extern __printf(3, 0)
+int ls_vsnprintf(char *buf, size_t size, const char *fmt, va_list args);
+#endif
 extern __printf(3, 4)
 int scnprintf(char *buf, size_t size, const char *fmt, ...);
 extern __printf(3, 0)
