diff --git a/include/linux/compiler_types.h b/include/linux/compiler_types.h
index b6ff83a714ca..3d807a004b6d 100644
--- a/include/linux/compiler_types.h
+++ b/include/linux/compiler_types.h
@@ -32,10 +32,18 @@ static inline void __chk_io_ptr(const volatile void __iomem *ptr) { }
 # ifdef STRUCTLEAK_PLUGIN
 #  define __user	__attribute__((user))
 # else
+#ifdef CONFIG_HAKC
+# define __user		__attribute__((kernel_user_ptr))
+#else
 #  define __user
+#endif
 # endif
 # define __iomem
+#ifdef CONFIG_HAKC
+# define __percpu	__attribute__((per_cpu_ptr))
+#else
 # define __percpu
+#endif
 # define __rcu
 # define __chk_user_ptr(x)	(void)0
 # define __chk_io_ptr(x)	(void)0
