diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index e8a807c78110..b82407ca5d9d 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -297,6 +297,11 @@ static int vmap_range_noflush(unsigned long addr, unsigned long end,
 
 	start = addr;
 	pgd = pgd_offset_k(addr);
+
+#if IS_ENABLED(CONFIG_HAKC_ARM_V9)
+	prot = __pgprot(pgprot_val(prot) | PROT_NORMAL_TAGGED);
+#endif
+
 	do {
 		next = pgd_addr_end(addr, end);
 		err = vmap_p4d_range(pgd, addr, next, phys_addr, prot,
@@ -2987,6 +2992,13 @@ void *__vmalloc_node_range(unsigned long size, unsigned long align,
 		return NULL;
 	}
 
+#if IS_ENABLED(CONFIG_HAKC_ARM_V9)
+	if (gfp_mask & GFP_KERNEL) {
+		prot = __pgprot(pgprot_val(prot) |
+				pgprot_val(PAGE_KERNEL_TAGGED));
+        }
+#endif
+
 	if (vmap_allow_huge && !(vm_flags & VM_NO_HUGE_VMAP)) {
 		unsigned long size_per_node;
 
